--- gdm-2.14.1/config/PostLogin.old	2006-04-12 17:06:08.000000000 -0400
+++ gdm-2.14.1/config/PostLogin	2006-04-12 17:07:18.000000000 -0400
@@ -1,8 +1,12 @@
 #!/bin/sh
 #
-# Note: this is a sample and will not be run as is.  Change the name of this
-# file to <gdmconfdir>/PostLogin/Default for this script to be run.  This
-# script will be run before any setup is run on behalf of the user and is
+# This script will be run before any setup is run on behalf of the user and is
 # useful if you for example need to do some setup to create a home directory
 # for the user or something like that.  $HOME, $LOGIN and such will all be
 # set appropriately and this script is run as root.
+
+if [ -x /usr/bin/dbus-daemon ]
+then
+	/usr/bin/dbus-daemon --session --fork
+fi
+
