--- gdm-2.8.0.4/config/gdm.conf.in.orig	2005-10-02 12:30:11.000000000 -0400
+++ gdm-2.8.0.4/config/gdm.conf.in	2005-10-02 12:32:57.000000000 -0400
@@ -45,7 +45,7 @@
 
 # The greeter for local (non-xdmcp) logins.  Change gdmlogin to gdmgreeter to
 # get the new graphical greeter.
-#Greeter=@EXPANDED_LIBEXECDIR@/gdmlogin
+Greeter=@EXPANDED_LIBEXECDIR@/gdmgreeter
 
 # The greeter for xdmcp logins, usually you want a less graphically intensive
 # greeter here so it's better to leave this with gdmlogin
@@ -253,7 +253,7 @@
 #GtkRC=@EXPANDED_DATADIR@/themes/Default/gtk-2.0/gtkrc
 
 # The GTK+ theme to use for the gui
-#GtkTheme=Default
+GtkTheme=Clearlooks
 # If to allow changing the GTK+ (widget) theme from the greeter.  Currently
 # this only affects the standard greeter as the graphical greeter does
 # not yet have this ability
@@ -310,7 +310,7 @@
 #Logo=@EXPANDED_PIXMAPDIR@/gdm-foot-logo.png
 # The standard greeter should shake if a user entered the wrong username or
 # password.  Kind of cool looking
-#Quiver=true
+Quiver=true
 # The Actions menu (formerly system menu) is shown in the greeter, this is the
 # menu that contains reboot, shutdown, suspend, config and chooser.  None of
 # these is available if this is off.  They can be turned off individually
@@ -385,7 +385,7 @@
 # then provide a list that is delimited by /: to the GraphicalThemes key and 
 # set GraphicalThemeRand to true.  Otherwise use GraphicalTheme and specify
 # just one theme.
-#GraphicalTheme=circles
+GraphicalTheme=frg_gdm
 #GraphicalThemes=circles/:happygnome
 GraphicalThemeDir=@EXPANDED_DATADIR@/gdm/themes/
 GraphicalThemeRand=false
