--- gdm-2.6.0.8/config/gdm.conf.in.orig	2005-03-14 18:23:53.000000000 -0500
+++ gdm-2.6.0.8/config/gdm.conf.in	2005-03-14 18:31:48.000000000 -0500
@@ -45,7 +45,7 @@
 
 # The greeter for local (non-xdmcp) logins.  Change gdmlogin to gdmgreeter to
 # get the new graphical greeter.
-#Greeter=@EXPANDED_BINDIR@/gdmlogin
+Greeter=@EXPANDED_BINDIR@/gdmgreeter
 
 # The greeter for xdmcp logins, usually you want a less graphically intensive
 # greeter here so it's better to leave this with gdmlogin
@@ -243,7 +243,7 @@
 #GtkRC=@EXPANDED_DATADIR@/themes/Default/gtk/gtkrc
 
 # The GTK+ theme to use for the gui
-#GtkTheme=Default
+GtkTheme=Clearlooks
 # If to allow changing the GTK+ (widget) theme from the greeter.  Currently
 # this only affects the standard greeter as the graphical greeter does
 # not yet have this ability
@@ -282,7 +282,7 @@
 #Logo=@EXPANDED_PIXMAPDIR@/gdm-foot-logo.png
 # The standard greeter should shake if a user entered the wrong username or
 # password.  Kind of cool looking
-#Quiver=true
+Quiver=true
 # The Actions menu (formerly system menu) is shown in the greeter, this is the
 # menu that contains reboot, shutdown, suspend, config and chooser.  None of
 # these is available if this is off.  They can be turned off individually
@@ -349,7 +349,7 @@
 #UseInvisibleInEntry=false
 # These two keys are for the new greeter.  Circles is the standard
 # shipped theme
-#GraphicalTheme=circles
+GraphicalTheme=frg_gdm
 GraphicalThemeDir=@EXPANDED_DATADIR@/gdm/themes/
 # If InfoMsgFile points to a file, the greeter will display the contents of the
 # file in a modal dialog box before the user is allowed to log in.
