--- rhythmbox-0.9.5/metadata/rb-metadata-dbus-client.c.orig	2006-09-15 04:31:42.000000000 -0400
+++ rhythmbox-0.9.5/metadata/rb-metadata-dbus-client.c	2006-09-15 04:32:44.000000000 -0400
@@ -118,7 +118,7 @@
 	if (dbus_connection) {
 		if (dbus_connection_get_is_connected (dbus_connection)) {
 			rb_debug ("closing dbus connection");
-			dbus_connection_disconnect (dbus_connection);
+			dbus_connection_close (dbus_connection);
 		} else {
 			rb_debug ("dbus connection already closed");
 		}
--- rhythmbox-0.9.5/metadata/rb-metadata-dbus-service.c.orig	2006-09-15 04:31:56.000000000 -0400
+++ rhythmbox-0.9.5/metadata/rb-metadata-dbus-service.c	2006-09-15 04:33:09.000000000 -0400
@@ -494,7 +494,7 @@
 	g_main_loop_run (svc.loop);
 
 	if (svc.connection) {
-		dbus_connection_disconnect (svc.connection);
+		dbus_connection_close (svc.connection);
 		dbus_connection_unref (svc.connection);
 	}
 
