diff -Naur balsa-2.3.27.old/libbalsa/imap/imap-commands.c balsa-2.3.27/libbalsa/imap/imap-commands.c
--- balsa-2.3.27.old/libbalsa/imap/imap-commands.c	2008-12-30 11:11:20.000000000 +0000
+++ balsa-2.3.27/libbalsa/imap/imap-commands.c	2009-01-11 16:39:51.000000000 +0000
@@ -1570,7 +1570,6 @@
   if(imap_mbox_handle_can_do(h, IMCAP_THREAD_REFERENCES)) {
     int can_do_literals = imap_mbox_handle_can_do(h, IMCAP_LITERAL);
     unsigned cmdno;
-    ImapResponse rc;
     ImapCmdTag tag;
 
     cmdno = imap_make_tag(tag);
