diff -Nur gdm-2.19.5/gui/gdmsetup.c gdm-2.19.5.new/gui/gdmsetup.c
--- gdm-2.19.5/gui/gdmsetup.c	2007-07-31 10:25:49.000000000 +0200
+++ gdm-2.19.5.new/gui/gdmsetup.c	2007-07-31 10:25:50.000000000 +0200
@@ -2128,20 +2128,15 @@
 
 	if (strcmp (ve_sure_string (name), "sysmenu") == 0) {
 	
-		GtkWidget *config_available;
 		GtkWidget *chooser_button;
 		
-		config_available = glade_xml_get_widget (xml, "config_available");
 		chooser_button = glade_xml_get_widget (xml, "chooser_button");
 
-		gtk_widget_set_sensitive (config_available, val);
 		gtk_widget_set_sensitive (chooser_button, val);
 		
 		g_signal_connect (G_OBJECT (toggle), "toggled",
 		                  G_CALLBACK (toggle_toggled), toggle);	
 		g_signal_connect (G_OBJECT (toggle), "toggled",
-		                  G_CALLBACK (toggle_toggled_sensitivity_positive), config_available);
-		g_signal_connect (G_OBJECT (toggle), "toggled",
 		                  G_CALLBACK (toggle_toggled_sensitivity_positive), chooser_button);
 	}
 	else if (strcmp ("autologin", ve_sure_string (name)) == 0) {
@@ -7336,7 +7331,6 @@
 {
 	/* Initialize and hookup callbacks for plain menu bar settings */
 	setup_notify_toggle ("sysmenu", GDM_KEY_SYSTEM_MENU);
-	setup_notify_toggle ("config_available", GDM_KEY_CONFIG_AVAILABLE);
 	setup_notify_toggle ("chooser_button", GDM_KEY_CHOOSER_BUTTON);
 }
 
diff -Nur gdm-2.19.5/gui/gdmsetup.glade gdm-2.19.5.new/gui/gdmsetup.glade
--- gdm-2.19.5/gui/gdmsetup.glade	2007-07-30 20:51:02.000000000 +0200
+++ gdm-2.19.5.new/gui/gdmsetup.glade	2007-07-31 10:25:50.000000000 +0200
@@ -1925,64 +1925,6 @@
 				  </child>
 
 				  <child>
-				    <widget class="GtkHBox" id="hbox105">
-				      <property name="visible">True</property>
-				      <property name="homogeneous">False</property>
-				      <property name="spacing">0</property>
-
-				      <child>
-					<widget class="GtkLabel" id="label171">
-					  <property name="visible">True</property>
-					  <property name="label">    </property>
-					  <property name="use_underline">False</property>
-					  <property name="use_markup">False</property>
-					  <property name="justify">GTK_JUSTIFY_LEFT</property>
-					  <property name="wrap">False</property>
-					  <property name="selectable">False</property>
-					  <property name="xalign">0.5</property>
-					  <property name="yalign">0.5</property>
-					  <property name="xpad">0</property>
-					  <property name="ypad">0</property>
-					  <property name="ellipsize">PANGO_ELLIPSIZE_NONE</property>
-					  <property name="width_chars">-1</property>
-					  <property name="single_line_mode">False</property>
-					  <property name="angle">0</property>
-					</widget>
-					<packing>
-					  <property name="padding">0</property>
-					  <property name="expand">False</property>
-					  <property name="fill">False</property>
-					</packing>
-				      </child>
-
-				      <child>
-					<widget class="GtkCheckButton" id="config_available">
-					  <property name="visible">True</property>
-					  <property name="tooltip" translatable="yes">Allows the configurator to be run from the greeter. Note that the user will need to type in the root password before the configurator will be started.</property>
-					  <property name="can_focus">True</property>
-					  <property name="label" translatable="yes">Include Con_figure menu item</property>
-					  <property name="use_underline">True</property>
-					  <property name="relief">GTK_RELIEF_NORMAL</property>
-					  <property name="focus_on_click">True</property>
-					  <property name="active">False</property>
-					  <property name="inconsistent">False</property>
-					  <property name="draw_indicator">True</property>
-					</widget>
-					<packing>
-					  <property name="padding">0</property>
-					  <property name="expand">False</property>
-					  <property name="fill">False</property>
-					</packing>
-				      </child>
-				    </widget>
-				    <packing>
-				      <property name="padding">0</property>
-				      <property name="expand">True</property>
-				      <property name="fill">True</property>
-				    </packing>
-				  </child>
-
-				  <child>
 				    <widget class="GtkHBox" id="hbox106">
 				      <property name="visible">True</property>
 				      <property name="homogeneous">False</property>
