--- daemon/verify-shadow.c.old	2008-01-25 15:06:36.000000000 +0000
+++ daemin/verify-shadow.c	2008-01-25 15:06:49.000000000 +0000
@@ -199,7 +199,7 @@
 
 	if (pwent == NULL) {
 		gdm_sleep_no_signal (gdm_daemon_config_get_value_int (GDM_KEY_RETRY_DELAY));
-		gdm_debug ("Couldn't authenticate user");
+		g_debug ("Couldn't authenticate user");
 
 		print_cant_auth_errbox ();
 
@@ -213,7 +213,7 @@
 	if (ppasswd == NULL || (ppasswd[0] != '\0' &&
 				strcmp (crypt (passwd, ppasswd), ppasswd) != 0)) {
 		gdm_sleep_no_signal (gdm_daemon_config_get_value_int (GDM_KEY_RETRY_DELAY));
-		gdm_debug ("Couldn't authenticate user");
+		g_debug ("Couldn't authenticate user");
 
 		print_cant_auth_errbox ();
 
@@ -227,7 +227,7 @@
 	    ( ! gdm_daemon_config_get_value_bool (GDM_KEY_ALLOW_REMOTE_ROOT) &&
               ! d->attached)) && pwent->pw_uid == 0) {
 
-		gdm_debug ("Root login disallowed on display '%s'", d->name);
+		g_debug ("Root login disallowed on display '%s'", d->name);
 		gdm_slave_greeter_ctl_no_ret (GDM_ERRBOX,
 					      _("The system administrator "
 						"is not allowed to login "
@@ -245,7 +245,7 @@
 	/* Check with the 'loginrestrictions' function
 	   if the user has been disallowed */
 	if (loginrestrictions (login, 0, NULL, &message) != 0) {
-		gdm_debug ("User not allowed to log in");
+		g_debug ("User not allowed to log in");
 		gdm_slave_greeter_ctl_no_ret (GDM_ERRBOX,
 					      _("\nThe system administrator "
 						"has disabled your "
@@ -269,7 +269,7 @@
 	    (strcmp (pwent->pw_shell, NOLOGIN) == 0 ||
 	     strcmp (pwent->pw_shell, "/bin/true") == 0 ||
 	     strcmp (pwent->pw_shell, "/bin/false") == 0)) {
-		gdm_debug ("User not allowed to log in");
+		g_debug ("User not allowed to log in");
 		gdm_slave_greeter_ctl_no_ret (GDM_ERRBOX,
 					      _("\nThe system administrator "
 						"has disabled your "
@@ -294,7 +294,7 @@
 	}
 
 	if ( ! gdm_setup_gids (login, pwent->pw_gid)) {
-		gdm_debug ("Cannot set user group");
+		g_debug ("Cannot set user group");
 		gdm_slave_greeter_ctl_no_ret (GDM_ERRBOX,
 					      _("\nCannot set your user group; "
 						"you will not be able to log in. "
@@ -307,7 +307,7 @@
 
 	switch (passwdexpired (login, &info_msg)) {
 	case 1 :
-		gdm_debug ("User password has expired");
+		g_debug ("User password has expired");
 		gdm_errorgui_error_box (d, GTK_MESSAGE_ERROR,
 					_("You are required to change your password.\n"
 					  "Please choose a new one."));
@@ -381,7 +381,7 @@
 		break;
 
 	case 2 :
-		gdm_debug ("User password has expired");
+		g_debug ("User password has expired");
 		gdm_errorgui_error_box (d, GTK_MESSAGE_ERROR,
 					_("Your password has expired.\n"
 					  "Only a system administrator can now change it"));
@@ -390,7 +390,7 @@
 		break;
 
 	case -1 :
-		gdm_debug ("Internal error on passwdexpired");
+		g_debug ("Internal error on passwdexpired");
 		gdm_errorgui_error_box (d, GTK_MESSAGE_ERROR,
 					_("An internal error occurred. You will not be able to log in.\n"
 					  "Please try again later or contact your system administrator."));
@@ -427,12 +427,12 @@
 
 	pwent = getpwnam (login);
 	if (pwent == NULL) {
-		gdm_debug ("Cannot get passwd structure for user");
+		g_debug ("Cannot get passwd structure for user");
 		return FALSE;
 	}
 
 	if ( ! gdm_setup_gids (login, pwent->pw_gid)) {
-		gdm_debug ("Cannot set user group");
+		g_debug ("Cannot set user group");
 		gdm_errorgui_error_box (d,
 					GTK_MESSAGE_ERROR,
 					_("\nCannot set your user group; "
