diff -Naur gtksourceview-sharp-2.0-0.12.old/Makefile.am gtksourceview-sharp-2.0-0.12/Makefile.am
--- gtksourceview-sharp-2.0-0.12.old/Makefile.am	2008-02-26 23:30:05.000000000 +0000
+++ gtksourceview-sharp-2.0-0.12/Makefile.am	2008-08-28 17:23:52.000000000 +0100
@@ -1,5 +1,5 @@
 
-SUBDIRS=		gtksourceview sample doc 
+SUBDIRS=		gtksourceview doc 
 DISTCLEANFILES = gtksourceview-sharp-2.0.pc
 EXTRA_DIST=		nemerle.lang vbnet.lang gtksourceview-sharp-2.0.pc.in gtk-sharp.snk
 
diff -Naur gtksourceview-sharp-2.0-0.12.old/Makefile.in gtksourceview-sharp-2.0-0.12/Makefile.in
--- gtksourceview-sharp-2.0-0.12.old/Makefile.in	2008-02-26 23:30:10.000000000 +0000
+++ gtksourceview-sharp-2.0-0.12/Makefile.in	2008-08-28 17:23:57.000000000 +0100
@@ -185,7 +185,7 @@
 target_vendor = @target_vendor@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = gtksourceview sample doc 
+SUBDIRS = gtksourceview doc 
 DISTCLEANFILES = gtksourceview-sharp-2.0.pc
 EXTRA_DIST = nemerle.lang vbnet.lang gtksourceview-sharp-2.0.pc.in gtk-sharp.snk
 extra_langdir = $(gtksourceview_prefix)/share/gtksourceview-1.0/language-specs
diff -Naur gtksourceview-sharp-2.0-0.12.old/configure gtksourceview-sharp-2.0-0.12/configure
--- gtksourceview-sharp-2.0-0.12.old/configure	2008-02-26 23:30:10.000000000 +0000
+++ gtksourceview-sharp-2.0-0.12/configure	2008-08-28 17:24:22.000000000 +0100
@@ -3549,7 +3549,7 @@
 fi
 
 
-ac_config_files="$ac_config_files Makefile gtksourceview/AssemblyInfo.cs gtksourceview/gtksourceview-sharp.dll.config gtksourceview/Makefile gtksourceview-sharp-2.0.pc doc/Makefile sample/Makefile"
+ac_config_files="$ac_config_files Makefile gtksourceview/AssemblyInfo.cs gtksourceview/gtksourceview-sharp.dll.config gtksourceview/Makefile gtksourceview-sharp-2.0.pc doc/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -4140,7 +4140,6 @@
     "gtksourceview/Makefile") CONFIG_FILES="$CONFIG_FILES gtksourceview/Makefile" ;;
     "gtksourceview-sharp-2.0.pc") CONFIG_FILES="$CONFIG_FILES gtksourceview-sharp-2.0.pc" ;;
     "doc/Makefile") CONFIG_FILES="$CONFIG_FILES doc/Makefile" ;;
-    "sample/Makefile") CONFIG_FILES="$CONFIG_FILES sample/Makefile" ;;
 
   *) { { echo "$as_me:$LINENO: error: invalid argument: $ac_config_target" >&5
 echo "$as_me: error: invalid argument: $ac_config_target" >&2;}
diff -Naur gtksourceview-sharp-2.0-0.12.old/configure.in gtksourceview-sharp-2.0-0.12/configure.in
--- gtksourceview-sharp-2.0-0.12.old/configure.in	2008-02-26 23:30:05.000000000 +0000
+++ gtksourceview-sharp-2.0-0.12/configure.in	2008-08-28 17:24:36.000000000 +0100
@@ -122,7 +122,6 @@
 gtksourceview/Makefile
 gtksourceview-sharp-2.0.pc
 doc/Makefile
-sample/Makefile
 ])
 
 echo "---"
