diff -Naur libexif-gtk-0.3.5.old/configure.in libexif-gtk-0.3.5/configure.in
--- libexif-gtk-0.3.5.old/configure.in	2004-10-18 20:12:58.000000000 +0100
+++ libexif-gtk-0.3.5/configure.in	2008-09-27 18:44:29.000000000 +0100
@@ -39,7 +39,7 @@
 dnl ---------------------------------------------------------------------------
 dnl libraries needed
 dnl ---------------------------------------------------------------------------
-PKG_CHECK_MODULES(LIBEXIF_GTK, gtk+-2.0 libexif >= 0.5.9)
+PKG_CHECK_MODULES(LIBEXIF_GTK, gtk+-2.0 libexif >= 0.6.12)
 AC_SUBST(LIBEXIF_GTK_LIBS)
 AC_SUBST(LIBEXIF_GTK_CFLAGS)
 PKG_CHECK_MODULES(GTK, gtk+-2.0)
diff -Naur libexif-gtk-0.3.5.old/gtk-extensions/Makefile.am libexif-gtk-0.3.5/gtk-extensions/Makefile.am
--- libexif-gtk-0.3.5.old/gtk-extensions/Makefile.am	2004-10-17 15:57:31.000000000 +0100
+++ libexif-gtk-0.3.5/gtk-extensions/Makefile.am	2008-09-27 18:44:37.000000000 +0100
@@ -1,7 +1,6 @@
 INCLUDES =				\
 	-I$(top_srcdir)			\
 	-I$(top_srcdir)/intl		\
-	-DGTK_DISABLE_DEPRECATED	\
 	$(GTK_CFLAGS)
 
 noinst_LTLIBRARIES = libgtk-extensions.la
diff -Naur libexif-gtk-0.3.5.old/gtk-extensions/Makefile.in libexif-gtk-0.3.5/gtk-extensions/Makefile.in
--- libexif-gtk-0.3.5.old/gtk-extensions/Makefile.in	2004-10-18 20:13:46.000000000 +0100
+++ libexif-gtk-0.3.5/gtk-extensions/Makefile.in	2008-09-27 18:44:46.000000000 +0100
@@ -219,7 +219,6 @@
 INCLUDES = \
 	-I$(top_srcdir)			\
 	-I$(top_srcdir)/intl		\
-	-DGTK_DISABLE_DEPRECATED	\
 	$(GTK_CFLAGS)
 
 noinst_LTLIBRARIES = libgtk-extensions.la
diff -Naur libexif-gtk-0.3.5.old/libexif-gtk/Makefile.am libexif-gtk-0.3.5/libexif-gtk/Makefile.am
--- libexif-gtk-0.3.5.old/libexif-gtk/Makefile.am	2004-10-17 14:48:35.000000000 +0100
+++ libexif-gtk-0.3.5/libexif-gtk/Makefile.am	2008-09-27 18:43:35.000000000 +0100
@@ -3,8 +3,7 @@
 	-I$(top_srcdir)/intl				\
 	-I$(top_srcdir)/gtk-extensions			\
 	$(LIBEXIF_GTK_CFLAGS)				\
-	-DG_LOG_DOMAIN=\"libexif\"			\
-	-DGTK_DISABLE_DEPRECATED
+	-DG_LOG_DOMAIN=\"libexif\"            
 
 lib_LTLIBRARIES = libexif-gtk.la
 
diff -Naur libexif-gtk-0.3.5.old/libexif-gtk/Makefile.in libexif-gtk-0.3.5/libexif-gtk/Makefile.in
--- libexif-gtk-0.3.5.old/libexif-gtk/Makefile.in	2004-10-18 20:13:46.000000000 +0100
+++ libexif-gtk-0.3.5/libexif-gtk/Makefile.in	2008-09-27 18:43:49.000000000 +0100
@@ -252,7 +252,6 @@
 	-I$(top_srcdir)/gtk-extensions			\
 	$(LIBEXIF_GTK_CFLAGS)				\
 	-DG_LOG_DOMAIN=\"libexif\"			\
-	-DGTK_DISABLE_DEPRECATED
 
 lib_LTLIBRARIES = libexif-gtk.la
 libexif_gtkincludedir = $(includedir)/libexif-gtk
