--- gnome-bluetooth-0.7.0.orig/debian/control
+++ gnome-bluetooth-0.7.0/debian/control
@@ -0,0 +1,30 @@
+Source: gnome-bluetooth
+Section: gnome
+Priority: optional
+Maintainer: Edd Dumbill <ejad@debian.org>
+Build-Depends: debhelper (>> 4.2.0), intltool, libbluetooth1-dev (>= 2.7-1), cdbs, python-gtk2-dev, pkg-config, libglib2.0-dev, libglade2-dev, libgnomeui-dev, python2.4-dev, gob2, libopenobex-1.0-0-dev, libbtctl2-dev (>= 0.6.0), librsvg2-dev, libgnomevfs2-dev (>= 2.1.3.1), libgtk2.0-dev (>= 2.4.0), liborbit2-dev, gconf2, libbonobo2-dev, libbonobo2-dev
+Standards-Version: 3.6.2
+
+Package: libgnomebt0
+Section: libs
+Depends: ${shlibs:Depends}
+Architecture: any
+Description: GNOME Bluetooth library
+ Libraries to support the GNOME Bluetooth subsystem.
+
+Package: libgnomebt0-dev
+Section: libdevel
+Provides: libgnomebt-dev
+Conflicts: libgnomebt-dev
+Depends: libgnomebt0 (= ${Source-Version})
+Architecture: any
+Description: Development files for GNOME Bluetooth library
+ Development files for working with the GNOME Bluetooth subsystem.
+
+Package: gnome-bluetooth
+Section: gnome
+Depends: ${shlibs:Depends}, ${python:Depends}, python2.4-libbtctl, python2.4-glade2, python2.4-gtk2, python2.4-gnome2
+Architecture: any
+Description: GNOME Bluetooth tools.
+ This package contains tools for managing and manipulating Bluetooth
+ devices using the GNOME desktop.
--- gnome-bluetooth-0.7.0.orig/debian/rules
+++ gnome-bluetooth-0.7.0/debian/rules
@@ -0,0 +1,7 @@
+#!/usr/bin/make -f
+include /usr/share/cdbs/1/rules/debhelper.mk
+include /usr/share/cdbs/1/class/gnome.mk
+include /usr/share/cdbs/1/rules/simple-patchsys.mk
+
+binary-install/gnome-bluetooth::
+	dh_python -pgnome-bluetooth
--- gnome-bluetooth-0.7.0.orig/debian/changelog
+++ gnome-bluetooth-0.7.0/debian/changelog
@@ -0,0 +1,134 @@
+gnome-bluetooth (0.7.0-0ubuntu2) dapper; urgency=low
+
+  * Rebuild to link against gnutls12 (unnecessary dependency, but --as-needed
+    breaks).
+
+ -- Martin Pitt <martin.pitt@ubuntu.com>  Thu,  9 Mar 2006 08:41:30 +0100
+
+gnome-bluetooth (0.7.0-0ubuntu1) dapper; urgency=low
+
+  * New upstream release:
+    - Fix a bad crasher that was plaguing 0.6.0 users
+    - Have the menu icons magically appear again
+    - Fix some memory leaks
+    - Better message when an error occurs sending a file to a Palm
+    - Fix build warnings
+    - Fix Python detection for bi-arch systems
+    - Update for libbtctl 0.6.0.
+  * debian/control:
+    - bumped Build-Dep on libbtctl2-dev (>= 0.6.0).  
+    - replaced libbonobo-activation-dev by libbonobo2-dev
+  * debian/rules:
+    - removed unnecessary build options.
+    - use gnome.mk, drop buildcore.mk, autotools.mk.
+  * debian/patches/fix-python-Makefile.patch:
+    - fixed buildsystem to install python files
+  * debian/patches/fix-python-Makefile.patch:
+    - fixed buildsystem to install python files.
+  * debian/patches/MenusRevisited.patch:
+    - hide both menu entries as per MenusRevisited.
+
+ -- Daniel Holbach <daniel.holbach@ubuntu.com>  Thu, 24 Nov 2005 18:26:41 +0100
+
+gnome-bluetooth (0.6.0+cvs2005.09.09-0ubuntu1) breezy; urgency=low
+
+  * CVS fixes:
+    - src/fileactiondialog.gob: don't use uri after it was freed, make
+      sure that a URI was created before using it
+    - src/gnomebt-fileactiondialog-test.c: (main): check that the
+      files exist before running the tests
+    - src/controller.gob: fix 2 memory leaks, gconf_client_get_string
+      already returns a newly-allocated string
+    - src/permissiondialog.gob: fix a double-free
+  * fixes Malone #2406.
+
+ -- Daniel Holbach <dh@mailempfang.de>  Tue, 20 Sep 2005 19:36:24 +0200
+
+gnome-bluetooth (0.6.0-0ubuntu1) breezy; urgency=low
+
+  * New upstream release:
+    - gnome-obex-server: Now preserves timestamps of reserved files 
+      (Tumoas Salo).
+    - Save received files to the path specified in the
+      /apps/gnome-obex-server/savedir GConf key
+    - Remove the nautilus extension, sending files over Bluetooth 
+      devices is now handled though nautilus-sendto (Bastien Nocera)
+    - Make gnome-bluetooth-manager run with newer PyGTK (Olav Vitters,
+      Harald Hoyer)
+    - Make dialogues comply better with the HIG (Bastien Nocera)
+    - Add a test program for the spinner widget (Bastien Nocera)
+    - Build fixes (Bastien Nocera)
+    - Updated translations.
+  * debian/control:
+    - Bumped Standards-Version.
+    - Adjusted Build-Depends according to configure.in
+  * debian/patches/01_gnome-ui-appbar-fix.patch:
+    - dropped.
+
+ -- Daniel Holbach <dh@mailempfang.de>  Fri,  9 Sep 2005 17:12:18 +0200
+
+gnome-bluetooth (0.5.1-1ubuntu7) hoary; urgency=low
+
+  * Rebuilt, because of libhowl0 moving out.
+
+ -- Daniel Holbach <dh@mailempfang.de>  Thu, 10 Mar 2005 13:55:24 +0100
+
+gnome-bluetooth (0.5.1-1ubuntu6) hoary; urgency=low
+
+  * debian/patches/01_gnome-ui-appbar-fix.patch:
+    - gtk.Statusbar() fix thanks very much to Trent Lloyd.
+  * debian/control:
+    - Build-Depend on librsvg2-dev, python2.4-dev. Migrate all to python2.4.
+
+ -- Jeff Waugh <jeff.waugh@ubuntu.com>  Mon, 24 Jan 2005 19:11:43 +1100
+
+gnome-bluetooth (0.5.1-1ubuntu5) hoary; urgency=low
+
+  * debian/control:
+    - Build-Depends typo.
+
+ -- Jeff Waugh <jeff.waugh@canonical.com>  Thu,  2 Dec 2004 21:51:04 +1100
+
+gnome-bluetooth (0.5.1-1ubuntu4) hoary; urgency=low
+
+  * debian/control:
+    - Eugh, Build-Depends.
+
+ -- Jeff Waugh <jeff.waugh@canonical.com>  Thu,  2 Dec 2004 15:05:29 +1100
+
+gnome-bluetooth (0.5.1-1ubuntu3) hoary; urgency=low
+
+  * debian/control:
+    - Build-Depend on intltool
+
+ -- Jeff Waugh <jeff.waugh@canonical.com>  Thu,  2 Dec 2004 14:12:48 +1100
+
+gnome-bluetooth (0.5.1-1ubuntu2) hoary; urgency=low
+
+  * Upload to Ubuntu
+
+ -- Jeff Waugh <jeff.waugh@canonical.com>  Wed,  1 Dec 2004 18:47:26 +1100
+
+gnome-bluetooth (0.5.1-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Edd Dumbill <ejad@debian.org>  Fri, 18 Jun 2004 20:13:52 +0100
+
+gnome-bluetooth (0.5-3) unstable; urgency=low
+
+  * Add more missing dependencies for gnome-bluetooth (gtk2, glade2, gnome2).
+
+ -- Edd Dumbill <ejad@debian.org>  Sat, 12 Jun 2004 02:54:48 +0100
+
+gnome-bluetooth (0.5-2) unstable; urgency=low
+
+  * Add missing dependency on python2.3-libbtctl. 
+
+ -- Edd Dumbill <ejad@debian.org>  Fri, 11 Jun 2004 15:22:36 +0100
+
+gnome-bluetooth (0.5-1) unstable; urgency=low
+
+  * Initial packaging.
+
+ -- Edd Dumbill <ejad@debian.org>  Thu, 10 Jun 2004 11:07:33 +0100
--- gnome-bluetooth-0.7.0.orig/debian/libgnomebt0.shlibs
+++ gnome-bluetooth-0.7.0/debian/libgnomebt0.shlibs
@@ -0,0 +1 @@
+libgnomebt 0 libgnomebt0 (>= 0.5.1)
--- gnome-bluetooth-0.7.0.orig/debian/compat
+++ gnome-bluetooth-0.7.0/debian/compat
@@ -0,0 +1 @@
+4
--- gnome-bluetooth-0.7.0.orig/debian/gnome-bluetooth.install
+++ gnome-bluetooth-0.7.0/debian/gnome-bluetooth.install
@@ -0,0 +1,9 @@
+debian/tmp/usr/bin/*
+debian/tmp/usr/lib/python2.4/site-packages/gnomebt/*.so.*
+debian/tmp/usr/lib/python2.4/site-packages/gnomebt/*.so
+debian/tmp/usr/lib/python2.4/site-packages/gnomebt/*.py
+debian/tmp/usr/share/gnome-bluetooth/*
+debian/tmp/usr/share/applications/*
+debian/tmp/usr/lib/bonobo/servers/*
+debian/tmp/usr/lib/bonobo/*.so
+debian/tmp/usr/share/locale/*
--- gnome-bluetooth-0.7.0.orig/debian/libgnomebt0-dev.install
+++ gnome-bluetooth-0.7.0/debian/libgnomebt0-dev.install
@@ -0,0 +1,5 @@
+debian/tmp/usr/include/*
+debian/tmp/usr/lib/lib*.so
+debian/tmp/usr/lib/lib*.la
+debian/tmp/usr/lib/lib*.a
+debian/tmp/usr/lib/pkgconfig/*
--- gnome-bluetooth-0.7.0.orig/debian/patches/fix-python-Makefile.patch
+++ gnome-bluetooth-0.7.0/debian/patches/fix-python-Makefile.patch
@@ -0,0 +1,47 @@
+diff -Nur gnome-bluetooth-0.7.0/python/Makefile.am gnome-bluetooth-0.7.0.new/python/Makefile.am
+--- gnome-bluetooth-0.7.0/python/Makefile.am	2005-11-24 14:22:03.000000000 +0100
++++ gnome-bluetooth-0.7.0.new/python/Makefile.am	2005-11-24 14:42:16.000000000 +0100
+@@ -1,6 +1,6 @@
+ 
+ pythondir = @pythondir@/gnomebt
+-pkgpythondir = $(pythondir)
++pkgpythondir = $(PYTHON_PREFIX)/lib/python$(PYTHON_VERSION)/site-packages/gnomebt
+ libdir = $(pkgpythondir)
+ 
+ python_LTLIBRARIES = iconlist.la
+diff -Nur gnome-bluetooth-0.7.0/python/Makefile.in gnome-bluetooth-0.7.0.new/python/Makefile.in
+--- gnome-bluetooth-0.7.0/python/Makefile.in	2005-11-21 23:23:21.000000000 +0100
++++ gnome-bluetooth-0.7.0.new/python/Makefile.in	2005-11-24 14:42:16.000000000 +0100
+@@ -258,7 +258,7 @@
+ prefix = @prefix@
+ program_transform_name = @program_transform_name@
+ pyexecdir = @pyexecdir@
+-pythondir = @pythondir@/gnomebt
++pythondir = $(PYTHON_PREFIX)/lib/python$(PYTHON_VERSION)/site-packages/gnomebt
+ sbindir = @sbindir@
+ sharedstatedir = @sharedstatedir@
+ subdirs = @subdirs@
+diff -Nur gnome-bluetooth-0.7.0/src/Makefile.am gnome-bluetooth-0.7.0.new/src/Makefile.am
+--- gnome-bluetooth-0.7.0/src/Makefile.am	2005-11-24 14:22:03.000000000 +0100
++++ gnome-bluetooth-0.7.0.new/src/Makefile.am	2005-11-24 14:42:16.000000000 +0100
+@@ -118,7 +118,7 @@
+ 		--override $(srcdir)/gnomebt-chooser.override \
+ 		$(srcdir)/gnomebt-chooser.defs > $@
+ 
+-pydir = @pythondir@/gnomebt/
++pydir = $(PYTHON_PREFIX)/lib/python$(PYTHON_VERSION)/site-packages/gnomebt/
+ 
+ controller_la_SOURCES = gnomebt-controller-py.c gnomebt-controller-pymodule.c
+ controller_la_LIBADD =  libgnomebt.la
+diff -Nur gnome-bluetooth-0.7.0/src/Makefile.in gnome-bluetooth-0.7.0.new/src/Makefile.in
+--- gnome-bluetooth-0.7.0/src/Makefile.in	2005-11-21 23:23:22.000000000 +0100
++++ gnome-bluetooth-0.7.0.new/src/Makefile.in	2005-11-24 14:42:40.000000000 +0100
+@@ -404,7 +404,7 @@
+ 
+ EXTRA_DIST = $(GOB_DIST) $(PYTHON_DIST)
+ CLEANFILES = $(PYTHON_CLEANFILES)
+-pydir = @pythondir@/gnomebt/
++pydir = $(PYTHON_PREFIX)/lib/python$(PYTHON_VERSION)/site-packages/gnomebt
+ controller_la_SOURCES = gnomebt-controller-py.c gnomebt-controller-pymodule.c
+ controller_la_LIBADD = libgnomebt.la
+ controller_la_LDFLAGS = $(GNOME_BLUETOOTH_PYTHON_LIBS) \
--- gnome-bluetooth-0.7.0.orig/debian/patches/fix-icon-location.patch
+++ gnome-bluetooth-0.7.0/debian/patches/fix-icon-location.patch
@@ -0,0 +1,24 @@
+diff -Nur gnome-bluetooth-0.7.0/python/gnome-bluetooth-manager.desktop.in gnome-bluetooth-0.7.0.new/python/gnome-bluetooth-manager.desktop.in
+--- gnome-bluetooth-0.7.0/python/gnome-bluetooth-manager.desktop.in	2005-11-22 12:17:48.000000000 +0100
++++ gnome-bluetooth-0.7.0.new/python/gnome-bluetooth-manager.desktop.in	2005-11-22 12:18:13.000000000 +0100
+@@ -2,7 +2,7 @@
+ Encoding=UTF-8
+ _Name=Bluetooth Manager
+ _Comment=Display Bluetooth devices and their properties
+-Icon=blueradio-48
++Icon=blueradio
+ Exec=gnome-bluetooth-manager
+ NoDisplay=True
+ Terminal=0
+diff -Nur gnome-bluetooth-0.7.0/python/manager.py gnome-bluetooth-0.7.0.new/python/manager.py
+--- gnome-bluetooth-0.7.0/python/manager.py	2005-11-22 12:05:13.000000000 +0100
++++ gnome-bluetooth-0.7.0.new/python/manager.py	2005-11-22 12:18:26.000000000 +0100
+@@ -99,7 +99,7 @@
+         self.window.set_default_size (400, 200)
+         self.window.connect ("delete_event", self.delete_event)
+         self.window.connect ("destroy", self.destroy)
+-	window_icon_name = self.image_file ("blueradio-48.png")
++	window_icon_name = self.image_file ("blueradio.png")
+ 	window_icon = gtk.gdk.pixbuf_new_from_file (window_icon_name)
+ 	self.window.set_icon (window_icon)
+ 
--- gnome-bluetooth-0.7.0.orig/debian/patches/MenusRevisited.patch
+++ gnome-bluetooth-0.7.0/debian/patches/MenusRevisited.patch
@@ -0,0 +1,22 @@
+diff -Nur gnome-bluetooth-0.7.0/obex/gnome-obex-server.desktop.in gnome-bluetooth-0.7.0.new/obex/gnome-obex-server.desktop.in
+--- gnome-bluetooth-0.7.0/obex/gnome-obex-server.desktop.in	2005-11-19 21:20:01.000000000 +0100
++++ gnome-bluetooth-0.7.0.new/obex/gnome-obex-server.desktop.in	2005-11-22 10:05:27.000000000 +0100
+@@ -4,6 +4,7 @@
+ _Comment=Receive files sent by Bluetooth devices
+ Icon=blueradio-48
+ Exec=gnome-obex-server
++NoDisplay=true
+ Terminal=0
+ Type=Application
+ Categories=Application;System;
+diff -Nur gnome-bluetooth-0.7.0/python/gnome-bluetooth-manager.desktop.in gnome-bluetooth-0.7.0.new/python/gnome-bluetooth-manager.desktop.in
+--- gnome-bluetooth-0.7.0/python/gnome-bluetooth-manager.desktop.in	2005-11-19 21:20:01.000000000 +0100
++++ gnome-bluetooth-0.7.0.new/python/gnome-bluetooth-manager.desktop.in	2005-11-22 10:04:52.000000000 +0100
+@@ -4,6 +4,7 @@
+ _Comment=Display Bluetooth devices and their properties
+ Icon=blueradio-48
+ Exec=gnome-bluetooth-manager
++NoDisplay=true
+ Terminal=0
+ Type=Application
+ Categories=Application;System;
--- gnome-bluetooth-0.7.0.orig/debian/copyright
+++ gnome-bluetooth-0.7.0/debian/copyright
@@ -0,0 +1,34 @@
+This package was first debianized by Edd Dumbill <ejad@debian.org>
+on Thu Jun 10 11:14:24 BST 2004.
+
+It was downloaded from http://usefulinc.com/software/gnome-bluetooth/.
+
+Upstream Author:
+
+Edd Dumbill <edd@usefulinc.com>
+
+Copyright:
+
+    gnome-bluetooth tools
+    Copyright (C) 2003-4 Edd Dumbill <edd@usefulinc.com>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+
+    gnome-bluetooth library
+    Copyright (C) 2003-4 Edd Dumbill <edd@usefulinc.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+
+On Debian GNU/Linux systems, the complete text of the GNU General
+Public License can be found in `/usr/share/common-licenses/GPL-2'.
+
+On Debian GNU/Linux systems, the complete text of the GNU Lesser General
+Public License can be found in `/usr/share/common-licenses/LGPL-2.1'.
--- gnome-bluetooth-0.7.0.orig/debian/libgnomebt0.install
+++ gnome-bluetooth-0.7.0/debian/libgnomebt0.install
@@ -0,0 +1 @@
+debian/tmp/usr/lib/lib*.so.*
