--- gnome-netstatus-2.12.0/src/netstatus-dialog.c.orig	2005-10-09 23:59:59.000000000 -0400
+++ gnome-netstatus-2.12.0/src/netstatus-dialog.c	2005-10-10 00:00:12.000000000 -0400
@@ -38,7 +38,7 @@
 #define NETWORK_CONFIG_TOOL_KEY NETWORK_CONFIG_TOOL_DIR "/config_tool"
 
 static const char *network_config_tools[] = {
-  "network-admin --configure %i",
+  "gnomesu network-admin --configure %i",
   "redhat-config-network",
   "system-control-network"
 };
