--- src/applet.c.no-session	2007-03-26 22:29:43.000000000 -0400
+++ src/applet.c	2007-03-26 22:30:14.000000000 -0400
@@ -308,11 +308,6 @@
 
       first_time = TRUE;
 
-      program = gnome_program_init ("fast-user-switch-applet", VERSION,
-				    LIBGNOME_MODULE, argc, argv,
-				    GNOME_PROGRAM_STANDARD_PROPERTIES,
-				    NULL);
-
       /* Do this here so it's only done once. */
       gtk_rc_parse_string ("style \"fast-user-switch-menubar-style\"\n"
 			   "{\n"
