--- system-tools-backends-1.9.5.1/Utils/Platform.pm.orig	2006-09-17 11:30:27.000000000 -0400
+++ system-tools-backends-1.9.5.1/Utils/Platform.pm	2006-09-17 11:31:23.000000000 -0400
@@ -81,6 +81,7 @@
   "slackware-10.0.0" => [ "Slackware", "10.0.0", "" ],
   "slackware-10.1.0" => [ "Slackware", "10.1.0", "" ],
   "slackware-10.2.0" => [ "Slackware", "10.2.0", "" ],
+  "slackware-11.0.0" => [ "Slackware", "11.0.0", "" ],
   "freebsd-4"       => [ "FreeBSD", "4", "" ],
   "freebsd-5"       => [ "FreeBSD", "5", "" ],
   "freebsd-6"       => [ "FreeBSD", "6", "" ],
@@ -138,6 +139,7 @@
      "slackware-10.0.0" => "slackware-9.1.0",
      "slackware-10.1.0" => "slackware-9.1.0",
      "slackware-10.2.0" => "slackware-9.1.0",
+     "slackware-11.0.0" => "slackware-9.1.0",
      "suse-9.1"         => "suse-9.0",
      "ubuntu-5.04"      => "debian-3.0",
      "ubuntu-5.10"      => "debian-3.0",
