Index: network-conf.in
===================================================================
RCS file: /cvs/system-tools-backends/system-tools-backends/network-conf.in,v
retrieving revision 1.115
diff -u -r1.115 network-conf.in
--- network-conf.in	27 Aug 2005 15:24:22 -0000	1.115
+++ network-conf.in	8 Sep 2005 17:10:43 -0000
@@ -74,7 +74,7 @@
               "pld-1.0", "pld-1.1", "pld-1.99",
               "conectiva-9", "conectiva-10",
               "vine-3.0", "vine-3.1",
-              "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0",
+              "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0", "slackware-10.2.0",
               "gentoo", "vlos-1.2", "freebsd-5", "freebsd-6");
 
 $description =<<"end_of_description;";
Index: network.pl.in
===================================================================
RCS file: /cvs/system-tools-backends/system-tools-backends/network.pl.in,v
retrieving revision 1.208
diff -u -r1.208 network.pl.in
--- network.pl.in	27 Aug 2005 15:01:45 -0000	1.208
+++ network.pl.in	8 Sep 2005 17:10:46 -0000
@@ -522,6 +522,7 @@
        "slackware-9.1.0" => "redhat-6.2",
        "slackware-10.0.0" => "redhat-6.2",
        "slackware-10.1.0" => "redhat-6.2",
+       "slackware-10.2.0" => "redhat-6.2",
        "gentoo"       => "debian-2.2",
        "vlos-1.2"     => "debian-2.2",
        "freebsd-5"    => "debian-2.2",
@@ -1804,6 +1805,7 @@
     "slackware-9.1.0" => \&gst_network_deb22_get_file,
     "slackware-10.0.0" => \&gst_network_deb22_get_file,
     "slackware-10.1.0" => \&gst_network_deb22_get_file,
+    "slackware-10.2.0" => \&gst_network_deb22_get_file,
     "gentoo"       => \&gst_network_deb22_get_file,
     "freebsd-5"    => \&gst_network_deb22_get_file,
     "freebsd-6"    => \&gst_network_deb22_get_file,
@@ -3907,6 +3909,7 @@
     "slackware-9.1.0" => "",
     "slackware-10.0.0" => "",
     "slackware-10.1.0" => "",
+    "slackware-10.2.0" => "",
     "gentoo"       => "",
     "vlos-1.2"     => "",
     "freebsd-5"    => "",
@@ -4063,6 +4066,7 @@
     "slackware-9.1.0" => "slackware-9.1.0",
     "slackware-10.0.0" => "slackware-9.1.0",
     "slackware-10.1.0" => "slackware-9.1.0",
+    "slackware-10.2.0" => "slackware-9.1.0",
     "gentoo"       => "gentoo",
     "vlos-1.2"     => "vlos-1.2",
     "freebsd-5"    => "freebsd-5",
@@ -4507,6 +4511,7 @@
     "slackware-9.1.0" => "slackware-9.1.0",
     "slackware-10.0.0" => "slackware-9.1.0",
     "slackware-10.1.0" => "slackware-9.1.0",
+    "slackware-10.2.0" => "slackware-9.1.0",
     "gentoo"       => "gentoo",
     "vlos-1.2"     => "vlos-1.2",
     "freebsd-5"    => "freebsd-5",
@@ -5381,6 +5386,7 @@
     "slackware-9.1.0" => "slackware-9.1.0",
     "slackware-10.0.0" => "slackware-9.1.0",
     "slackware-10.1.0" => "slackware-9.1.0",
+    "slackware-10.2.0" => "slackware-9.1.0",
     "gentoo"       => "gentoo",
     "vlos-1.2"     => "vlos-1.2",
     "freebsd-5"    => "freebsd-5",
@@ -5811,6 +5817,7 @@
     "slackware-9.1.0" => "slackware-9.1.0",
     "slackware-10.0.0" => "slackware-9.1.0",
     "slackware-10.1.0" => "slackware-9.1.0",
+    "slackware-10.2.0" => "slackware-9.1.0",
     "gentoo"       => "gentoo",
     "vlos-1.2"     => "vlos-1.2",
     "freebsd-5"    => "freebsd-5",
Index: platform.pl.in
===================================================================
RCS file: /cvs/system-tools-backends/system-tools-backends/platform.pl.in,v
retrieving revision 1.71
diff -u -r1.71 platform.pl.in
--- platform.pl.in	29 Jun 2005 22:35:20 -0000	1.71
+++ platform.pl.in	8 Sep 2005 17:10:46 -0000
@@ -74,6 +74,7 @@
   "slackware-9.1.0" => "Slackware 9.1.0",
   "slackware-10.0.0" => "Slackware 10.0.0",
   "slackware-10.1.0" => "Slackware 10.1.0",
+  "slackware-10.2.0" => "Slackware 10.2.0",
   "freebsd-4"       => "FreeBSD 4",
   "freebsd-5"       => "FreeBSD 5",
   "freebsd-6"       => "FreeBSD 6",
Index: service.pl.in
===================================================================
RCS file: /cvs/system-tools-backends/system-tools-backends/service.pl.in,v
retrieving revision 1.89
diff -u -r1.89 service.pl.in
--- service.pl.in	4 Jul 2005 22:24:25 -0000	1.89
+++ service.pl.in	8 Sep 2005 17:10:46 -0000
@@ -221,6 +221,7 @@
      "slackware-9.1.0" => "slackware-9.1.0",
      "slackware-10.0.0" => "slackware-9.1.0",
      "slackware-10.1.0" => "slackware-9.1.0",
+     "slackware-10.2.0" => "slackware-9.1.0",
 
      "gentoo"         => "gentoo",
      "vlos-1.2"       => "gentoo",
Index: services-conf.in
===================================================================
RCS file: /cvs/system-tools-backends/system-tools-backends/services-conf.in,v
retrieving revision 1.27
diff -u -r1.27 services-conf.in
--- services-conf.in	4 Jul 2005 22:24:25 -0000	1.27
+++ services-conf.in	8 Sep 2005 17:10:46 -0000
@@ -53,7 +53,7 @@
 
               "suse-7.0", "turbolinux-7.0", "fedora-1", "fedora-2", "fedora-3", "specifix",
               "pld-1.0", "pld-1.1", "pld-1.99",
-              "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0",
+              "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0", "slackware-10.2.0",
               "vine-3.0", "vine-3.1",
               "gentoo", "vlos-1.2", "freebsd-5", "freebsd-6", "suse-9.0", "suse-9.1");
 
Index: shares-conf.in
===================================================================
RCS file: /cvs/system-tools-backends/system-tools-backends/shares-conf.in,v
retrieving revision 1.90
diff -u -r1.90 shares-conf.in
--- shares-conf.in	21 Jun 2005 18:44:16 -0000	1.90
+++ shares-conf.in	8 Sep 2005 17:10:46 -0000
@@ -72,7 +72,7 @@
               "turbolinux-7.0", "fedora-1", "fedora-2", "fedora-3",
               "pld-1.0", "pld-1.1", "pld-1.99",
               "conectiva-9", "conectiva-10",
-              "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0",
+              "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0", "slackware-10.2.0",
               "gentoo", "vlos-1.2", "freebsd-5");
 
 $description =<<"end_of_description;";
@@ -553,6 +553,7 @@
      "slackware-9.1.0" => "slackware-9.1.0",
      "slackware-10.0.0" => "slackware-9.1.0",
      "slackware-10.1.0" => "slackware-9.1.0",
+     "slackware-10.2.0" => "slackware-9.1.0",
      "gentoo"       => "gentoo",
      "vlos-1.2"     => "gentoo",
      "freebsd-5"    => "freebsd-5",
Index: time-conf.in
===================================================================
RCS file: /cvs/system-tools-backends/system-tools-backends/time-conf.in,v
retrieving revision 1.120
diff -u -r1.120 time-conf.in
--- time-conf.in	21 Jun 2005 18:44:16 -0000	1.120
+++ time-conf.in	8 Sep 2005 17:10:46 -0000
@@ -68,7 +68,7 @@
               "mandrake-10.0", "mandrake-10.1",
               "debian-2.2", "debian-3.0", "debian-sarge",
               "suse-7.0", "suse-9.0", "suse-9.1", "turbolinux-7.0",
-              "slackware-8.0.0", "slackware-8.1", "slackware-9.0.0", "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0",
+              "slackware-8.0.0", "slackware-8.1", "slackware-9.0.0", "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0", "slackware-10.2.0",
               "gentoo", "vlos-1.2", "pld-1.0", "pld-1.1", "pld-1.99", "fedora-1", "fedora-2", "fedora-3", "specifix",
               "vine-3.0", "vine-3.1",
               "freebsd-5", "freebsd-6");
@@ -533,6 +533,7 @@
    "slackware-9.1.0" => "debian-2.2",
    "slackware-10.0.0" => "debian-2.2",
    "slackware-10.1.0" => "debian-2.2",
+   "slackware-10.2.0" => "debian-2.2",
 
    "gentoo"          => "gentoo",
    "vlos-1.2"        => "gentoo",
@@ -765,6 +766,7 @@
    "slackware-9.1.0" => "debian-2.2",
    "slackware-10.0.0" => "debian-2.2",
    "slackware-10.1.0" => "debian-2.2",
+   "slackware-10.2.0" => "debian-2.2",
 
    "gentoo"          => "gentoo",
    "vlos-1.2"        => "gentoo",
Index: users-conf.in
===================================================================
RCS file: /cvs/system-tools-backends/system-tools-backends/users-conf.in,v
retrieving revision 1.134
diff -u -r1.134 users-conf.in
--- users-conf.in	8 Aug 2005 18:04:12 -0000	1.134
+++ users-conf.in	8 Sep 2005 17:10:47 -0000
@@ -75,7 +75,7 @@
               "mandrake-10.0", "mandrake-10.1",
               "debian-2.2", "debian-3.0", "debian-sarge",
               "suse-7.0", "suse-9.0", "suse-9.1", "turbolinux-7.0",
-              "slackware-8.0.0", "slackware-8.1", "slackware-9.0.0", "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0",
+              "slackware-8.0.0", "slackware-8.1", "slackware-9.0.0", "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0", "slackware-10.2.0",
               "freebsd-4", "freebsd-5", "freebsd-6",
               "gentoo", "vlos-1.2",
               "archlinux-0.7",
@@ -370,6 +370,7 @@
   'slackware-9.1.0' => $gentoo_logindefs_defaults,
   'slackware-10.0.0' => $gentoo_logindefs_defaults,
   'slackware-10.1.0' => $gentoo_logindefs_defaults,
+  'slackware-10.2.0' => $gentoo_logindefs_defaults,
   'freebsd-4'       => $freebsd_logindefs_defaults,
   'freebsd-5'       => $freebsd_logindefs_defaults,
   'freebsd-6'       => $freebsd_logindefs_defaults,
