--- hal-0.5.2/fdi/policy/10osvendor/10-storage-policy.fdi.orig	2005-03-25 10:00:54.000000000 +0100
+++ hal-0.5.2/fdi/policy/10osvendor/10-storage-policy.fdi	2005-03-25 10:01:43.000000000 +0100
@@ -5,12 +5,12 @@
   <!-- Default policies merged onto computer root object  -->
   <device>
     <match key="info.udi" string="/org/freedesktop/Hal/devices/computer">
-      <merge key="storage.policy.default.mount_root" type="string">/media</merge>
-      <merge key="storage.policy.default.use_managed_keyword" type="bool">true</merge>
+      <merge key="storage.policy.default.mount_root" type="string">/mnt</merge>
+      <merge key="storage.policy.default.use_managed_keyword" type="bool">false</merge>
       <merge key="storage.policy.default.managed_keyword.primary" type="string">managed</merge>
       <merge key="storage.policy.default.managed_keyword.secondary" type="string">kudzu</merge>
       <merge key="storage.policy.default.mount_option.noauto" type="bool">true</merge>
-      <merge key="storage.policy.default.mount_option.pamconsole" type="bool">true</merge>
+      <merge key="storage.policy.default.mount_option.user" type="bool">true</merge>
       <merge key="storage.policy.default.mount_option.exec" type="bool">true</merge>
     </match>
   </device>


