--- hal.conf.orig	2006-07-26 20:04:12.000000000 +0200
+++ hal.conf	2006-08-14 17:25:51.000000000 +0200
@@ -49,5 +49,13 @@
     <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
   </policy>
 
+  <!-- Users who want/need HAL must be added to the "halusers" group -->
+  <policy group="62">
+    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
+    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+  </policy>
+
 </busconfig>
 
--- hal.conf.in.orig	2006-01-10 20:03:45.000000000 +0100
+++ hal.conf.in	2006-08-14 17:26:02.000000000 +0200
@@ -49,5 +49,13 @@
     <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
   </policy>
 
+  <!-- Users who want/need HAL must be added to the "halusers" group -->
+  <policy group="62">
+    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
+    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+  </policy>
+
 </busconfig>
 
