diff -Naur hal-0.5.2.orig/hald/linux2/probing/linux_dvd_rw_utils.h hal-0.5.2/hald/linux2/probing/linux_dvd_rw_utils.h
--- hal-0.5.2.orig/hald/linux2/probing/linux_dvd_rw_utils.h	2005-02-09 11:54:08.000000000 -0500
+++ hal-0.5.2/hald/linux2/probing/linux_dvd_rw_utils.h	2005-08-12 02:14:00.000000000 -0400
@@ -9,6 +9,8 @@
 #ifndef LINUX_DVD_RW_UTILS_H
 #define LINUX_DVD_RW_UTILS_H
 
+#define SG_FLAG_LUN_INHIBIT 2
+
 int get_dvd_r_rw_profile (int fd);
 int get_read_write_speed (int fd, int *read_speed, int *write_speed);
 int get_disc_type (int fd);
