diff -Naur f-spot-0.4.4.old/Makefile.in f-spot-0.4.4/Makefile.in
--- f-spot-0.4.4.old/Makefile.in	2008-05-28 20:38:03.000000000 +0100
+++ f-spot-0.4.4/Makefile.in	2008-05-29 18:17:27.000000000 +0100
@@ -892,8 +892,8 @@
 install-data-hook: 
 	$(UPDATE_DESKTOP)
 	if [ "x$(DIR_GIO)" != "x" ] ; then \
-		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll $(libdir)/f-spot/; \
-		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll.config $(libdir)/f-spot/; \
+		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll $(DESTDIR)/$(libdir)/f-spot/; \
+		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll.config $(DESTDIR)/$(libdir)/f-spot/; \
 	fi
 
 uninstall-hook: 
