--- metacity-2.10.0/src/metacity.schemas.in.orig	2005-03-14 17:17:41.000000000 -0500
+++ metacity-2.10.0/src/metacity.schemas.in	2005-03-14 17:18:41.000000000 -0500
@@ -114,7 +114,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Simple</default>
+      <default>Clearlooks</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
