diff -Naur ./configure ./configure
--- ./configure	2008-02-24 19:07:09.000000000 +0000
+++ ./configure	2008-03-27 18:48:45.000000000 +0000
@@ -44587,7 +44587,7 @@
 done
 
 
-for ac_header in ffmpeg/avformat.h
+for ac_header in libavformat/avformat.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -44727,7 +44727,7 @@
 done
 
 
-for ac_header in ffmpeg/avutil.h
+for ac_header in libavutil/avutil.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -44867,7 +44867,7 @@
 done
 
 
-for ac_header in postproc/postprocess.h
+for ac_header in libpostproc/postprocess.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -45003,8 +45003,8 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file postproc/postprocess.h." >&5
-echo "$as_me: error: Missing header file postproc/postprocess.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libpostproc/postprocess.h." >&5
+echo "$as_me: error: Missing header file libpostproc/postprocess.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
@@ -45243,7 +45243,7 @@
 done
 
 
-for ac_header in ffmpeg/avformat.h
+for ac_header in libavformat/avformat.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -45383,7 +45383,7 @@
 done
 
 
-for ac_header in ffmpeg/avutil.h
+for ac_header in libavutil/avutil.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -45523,7 +45523,7 @@
 done
 
 
-for ac_header in postproc/postprocess.h
+for ac_header in libpostproc/postprocess.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -45659,8 +45659,8 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file postproc/postprocess.h." >&5
-echo "$as_me: error: Missing header file postproc/postprocess.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libpostproc/postprocess.h." >&5
+echo "$as_me: error: Missing header file libpostproc/postprocess.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
@@ -46125,7 +46125,7 @@
 done
 
 
-for ac_header in ffmpeg/avformat.h
+for ac_header in libavformat/avformat.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -46265,7 +46265,7 @@
 done
 
 
-for ac_header in ffmpeg/avutil.h
+for ac_header in libavutil/avutil.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -46405,7 +46405,7 @@
 done
 
 
-for ac_header in postproc/postprocess.h
+for ac_header in libpostproc/postprocess.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -46541,8 +46541,8 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file postproc/postprocess.h." >&5
-echo "$as_me: error: Missing header file postproc/postprocess.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libpostproc/postprocess.h." >&5
+echo "$as_me: error: Missing header file libpostproc/postprocess.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
@@ -47008,7 +47008,7 @@
 done
 
 
-for ac_header in ffmpeg/avformat.h
+for ac_header in libavformat/avformat.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -47148,7 +47148,7 @@
 done
 
 
-for ac_header in ffmpeg/avutil.h
+for ac_header in libavutil/avutil.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -47288,7 +47288,7 @@
 done
 
 
-for ac_header in postproc/postprocess.h
+for ac_header in libpostproc/postprocess.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -47424,8 +47424,8 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file postproc/postprocess.h." >&5
-echo "$as_me: error: Missing header file postproc/postprocess.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libpostproc/postprocess.h." >&5
+echo "$as_me: error: Missing header file libpostproc/postprocess.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
diff -Naur ./modules/codec/ffmpeg/demux.c ./modules/codec/ffmpeg/demux.c
--- ./modules/codec/ffmpeg/demux.c	2007-11-21 22:23:29.000000000 +0000
+++ ./modules/codec/ffmpeg/demux.c	2008-03-27 18:52:20.000000000 +0000
@@ -35,7 +35,7 @@
 #ifdef HAVE_FFMPEG_AVFORMAT_H
 #   include <ffmpeg/avformat.h>
 #else
-#   include <avformat.h>
+#   include <libavformat/avformat.h>
 #endif
 
 #include "ffmpeg.h"
diff -Naur ./modules/codec/ffmpeg/mux.c ./modules/codec/ffmpeg/mux.c
--- ./modules/codec/ffmpeg/mux.c	2007-12-29 15:25:53.000000000 +0000
+++ ./modules/codec/ffmpeg/mux.c	2008-03-27 18:52:47.000000000 +0000
@@ -34,7 +34,7 @@
 #ifdef HAVE_FFMPEG_AVFORMAT_H
 #   include <ffmpeg/avformat.h>
 #else
-#   include <avformat.h>
+#   include <libavformat/avformat.h>
 #endif
 
 #include "ffmpeg.h"
