--- libfame-0.9.1.orig/src/fame_malloc.c	2003-06-20 08:40:30.000000000 -0400
+++ libfame-0.9.1/src/fame_malloc.c	2005-08-12 16:52:15.000000000 -0400
@@ -36,9 +36,11 @@
    */  
 
   ptr = (unsigned char*) malloc(size+ALIGN);
-  aligned = (unsigned char*) (((unsigned int)ptr & (~(ALIGN-1))) + ALIGN );
+  aligned = (unsigned char*) (((unsigned long)ptr &(~(ALIGN-1))) + ALIGN );
+  //  aligned = (unsigned char*) (((unsigned int)ptr & (~(ALIGN-1))) + ALIGN );
   padding = aligned - 1;
-  *padding = (ALIGN-1) - ((unsigned int)ptr & (ALIGN-1));
+  *padding = (ALIGN-1) - ((unsigned long)ptr & (ALIGN-1));
+  //  *padding = (ALIGN-1) - ((unsigned int)ptr & (ALIGN-1));
 
   return ((void*)aligned);
 }
