--- a/misc/devhelp.el~	2003-06-29 10:24:17.000000000 -0300
+++ b/misc/devhelp.el	2005-08-12 22:11:59.270346712 -0300
@@ -3,8 +3,7 @@
 (defun devhelp-word-at-point ()
   "runs devhelp"
   (interactive)
-  (setq w (current-word))
-  (start-process-shell-command "devhelp" nil "devhelp" "-s" w)
+  (start-process-shell-command "devhelp" nil "devhelp" "-s" (current-word))
   )
 
 ; Example: bind F7 to start devhelp and search for the word at the point.
