--- docs/reference/camel/Makefile.in.old	2008-01-09 06:39:06.000000000 +0000
+++ docs/reference/camel/Makefile.in	2008-01-09 06:39:16.000000000 +0000
@@ -684,8 +684,6 @@
 	  done; \
 	  echo '-- Installing $(srcdir)/html/index.sgml' ; \
 	  $(INSTALL_DATA) $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR) || :; \
-	  which gtkdoc-rebase >/dev/null && \
-	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
 	fi
 
 uninstall-local:
