diff -Naur etox-0.9.0.004.orig/src/style/Etox_Style.h etox-0.9.0.004/src/style/Etox_Style.h
--- etox-0.9.0.004.orig/src/style/Etox_Style.h	2004-07-29 22:58:28.000000000 -0400
+++ etox-0.9.0.004/src/style/Etox_Style.h	2005-09-21 20:58:09.000000000 -0400
@@ -10,6 +10,20 @@
 {
 #endif
 
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ewl/ewl_macros.h>
+
+#undef PATH_MAX
+#define PATH_MAX 4096
+
+#undef CHECK_PARAM_POINTER_RETURN
+#define CHECK_PARAM_POINTER_RETURN(sparam, param, ret) if (!(param)) { ecore_print_warning(__FUNCTION__, sparam); return ret; }
+
+#undef CHECK_PARAM_POINTER
+#define CHECK_PARAM_POINTER(sparam, param) if (!(param)) { ecore_print_warning(__FUNCTION__, sparam); return; }
+
 
 
 /*
diff -Naur etox-0.9.0.004.orig/src/style/etox_style.c etox-0.9.0.004/src/style/etox_style.c
--- etox-0.9.0.004.orig/src/style/etox_style.c	2005-01-28 21:48:38.000000000 -0500
+++ etox-0.9.0.004/src/style/etox_style.c	2005-09-21 18:05:43.000000000 -0400
@@ -1,3 +1,6 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include "etox_style_private.h"
 #include <ctype.h>
 #include <Evas.h>
diff -Naur etox-0.9.0.004.orig/src/style/etox_style_private.h etox-0.9.0.004/src/style/etox_style_private.h
--- etox-0.9.0.004.orig/src/style/etox_style_private.h	2004-10-11 22:54:16.000000000 -0400
+++ etox-0.9.0.004/src/style/etox_style_private.h	2005-09-21 18:03:57.000000000 -0400
@@ -2,6 +2,7 @@
 #define _ETOX_STYLE_PRIVATE_H
 
 #include <Ecore.h>
+#include <Ecore_Data.h>
 #include "Etox_Style.h"
 
 /*
diff -Naur etox-0.9.0.004.orig/src/style/etox_style_style.c etox-0.9.0.004/src/style/etox_style_style.c
--- etox-0.9.0.004.orig/src/style/etox_style_style.c	2004-07-29 22:58:28.000000000 -0400
+++ etox-0.9.0.004/src/style/etox_style_style.c	2005-09-21 20:50:43.000000000 -0400
@@ -1,5 +1,9 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include "../../config.h"
 #include "etox_style_private.h"
+#include <ewl/ewl_macros.h>
 
 #define SET_REL_COLOR(a, b) ((a + b) > 255 ? 255 : ((a + b) < 0 ? 0 : a + b))
 
diff -Naur etox-0.9.0.004.orig/test/selections.c etox-0.9.0.004/test/selections.c
--- etox-0.9.0.004.orig/test/selections.c	2004-05-18 20:04:17.000000000 -0400
+++ etox-0.9.0.004/test/selections.c	2005-09-21 20:58:28.000000000 -0400
@@ -1,3 +1,6 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include "../config.h"
 #include <Evas.h>
 #include <Ecore.h>
