diff -Naur control-center-2.11.91.orig/capplets/keyboard/Makefile.in control-center-2.11.91/capplets/keyboard/Makefile.in
--- control-center-2.11.91.orig/capplets/keyboard/Makefile.in	2005-08-10 06:36:21.000000000 -0400
+++ control-center-2.11.91/capplets/keyboard/Makefile.in	2005-08-22 10:39:51.000000000 -0400
@@ -243,7 +243,7 @@
 LIBEBOOK_CFLAGS = @LIBEBOOK_CFLAGS@
 LIBEBOOK_LIBS = @LIBEBOOK_LIBS@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lpangoxft-1.0
 LIBTOOL = @LIBTOOL@
 LIBXKLAVIER_CFLAGS = @LIBXKLAVIER_CFLAGS@
 LIBXKLAVIER_LIBS = @LIBXKLAVIER_LIBS@
diff -Naur control-center-2.11.91.orig/libkbdraw/Makefile.in control-center-2.11.91/libkbdraw/Makefile.in
--- control-center-2.11.91.orig/libkbdraw/Makefile.in	2005-08-10 06:36:24.000000000 -0400
+++ control-center-2.11.91/libkbdraw/Makefile.in	2005-08-22 10:38:43.000000000 -0400
@@ -55,7 +55,7 @@
 am_libkbdraw_a_OBJECTS = keyboard-marshal.$(OBJEXT) \
 	keyboard-drawing.$(OBJEXT)
 libkbdraw_a_OBJECTS = $(am_libkbdraw_a_OBJECTS)
-DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
+DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir) -I/usr/X11R6/include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
