--- ooo-build-2.4.1.7/patches/src680/apply	2008-06-24 12:55:42.000000000 +0100
+++ ooo-build-2.4.1.7.old/patches/src680/apply	2008-07-14 14:03:31.000000000 +0100
@@ -77,6 +77,9 @@
 # DroplineGNOME
 DroplineGNOME : LinuxCommon, DroplineGNOMEOnly, NotDebian, Mono, msaccess
 DroplineGNOME64 : LinuxCommon, DroplineGNOMEOnly, NotDebian, Mono, msaccess
+# GNOMESlackBuild
+GNOMESlackBuild : LinuxCommon, GNOMESlackBuildOnly, NotDebian, Mono, msaccess
+GNOMESlackBuild64 : LinuxCommon, GNOMESlackBuildOnly, NotDebian, Mono, msaccess
 # Ark Linux
 Ark: LinuxCommon, NotDebian, ArkOnly, BerkeleyDB43, Gcc43
 # www.pld-linux.org
@@ -975,6 +978,11 @@
 pushfont-psstream.diff, i#84481
 scp2-user-config-ooo-2.0.diff
 
+[ GNOMESlackBuildOnly ]
+SectionOwner => GNOMESlackBuild
+
+pushfont-psstream.diff, i#84481
+scp2-user-config-ooo-2.0.diff
 
 [ MandrivaOnly ]
 SectionOwner => gghibo
