--- src/service-discovery-config.in.old	2006-10-17 19:39:32.690776750 +0200
+++ src/service-discovery-config.in	2006-10-17 19:40:08.829035250 +0200
@@ -29,6 +29,8 @@
     gettext.bindtextdomain("@GETTEXT_PACKAGE@", "@LOCALEDIR@")
     gettext.textdomain("@GETTEXT_PACKAGE@")
     _ = gettext.gettext
+    import pygtk
+    pygtk.require('2.0')
     import gtk
     import gtk.glade
     gtk.glade.bindtextdomain("@GETTEXT_PACKAGE@", "@LOCALEDIR@")
--- tools/exec_wrapper.in.old	2006-10-17 19:40:02.700652250 +0200
+++ tools/exec_wrapper.in	2006-10-17 19:40:06.664900000 +0200
@@ -16,13 +16,13 @@
 import sys
 import os
 import pygtk
+pygtk.require('2.0')
 import gtk
 
 
 from subprocess import *
 from string import join, replace
 
-pygtk.require('2.0')
 
 def error_msg(msg):
     d = gtk.MessageDialog(parent=None, flags=gtk.DIALOG_MODAL,
