--- plugins/gconfterminal.py.in.old	2008-01-22 19:59:42.000000000 +0100
+++ plugins/gconfterminal.py.in	2008-01-22 19:59:55.000000000 +0100
@@ -34,7 +34,7 @@
 
 class plugin_gnometerminal:
     def __init__(self):
-        self.service_type = ["_ssh._tcp", "_sftp-ssh._tcp" ]
+        self.service_type = ["_ssh._tcp" ]
         self.author = "Sébastien Estienne"
 
     def enter_callback(self, widget, win):
