diff -Naur entrance-0.9.0.004.orig/src/client/entrance_ipc.h entrance-0.9.0.004/src/client/entrance_ipc.h
--- entrance-0.9.0.004.orig/src/client/entrance_ipc.h	2004-04-05 17:28:35.000000000 -0400
+++ entrance-0.9.0.004/src/client/entrance_ipc.h	2005-09-21 17:53:56.000000000 -0400
@@ -9,6 +9,7 @@
 #include <string.h>
 #include <limits.h>
 #include <sys/types.h>
+#include <sys/stat.h>
 #include <unistd.h>
 
 #include "../ipc.h"
diff -Naur entrance-0.9.0.004.orig/src/daemon/Entranced.h entrance-0.9.0.004/src/daemon/Entranced.h
--- entrance-0.9.0.004.orig/src/daemon/Entranced.h	2004-05-20 00:21:59.000000000 -0400
+++ entrance-0.9.0.004/src/daemon/Entranced.h	2005-09-21 17:55:09.000000000 -0400
@@ -13,6 +13,7 @@
 #include <syslog.h>
 
 #include <Ecore.h>
+#include <Ecore_Data.h>
 
 #include "../config.h"
 
