diff -Naur PolicyKit-0.9.old/src/polkit-dbus/polkit-resolve-exe-helper.c PolicyKit-0.9/src/polkit-dbus/polkit-resolve-exe-helper.c
--- PolicyKit-0.9.old/src/polkit-dbus/polkit-resolve-exe-helper.c	2008-05-30 22:24:44.000000000 +0100
+++ PolicyKit-0.9/src/polkit-dbus/polkit-resolve-exe-helper.c	2008-08-25 01:16:33.000000000 +0100
@@ -43,7 +43,6 @@
 #ifdef HAVE_FREEBSD
 #include <sys/param.h>
 #endif
-#include <security/pam_appl.h>
 #include <grp.h>
 #include <pwd.h>
 #include <syslog.h>
