--- balsa-2.3.4/images/Makefile.in.orig	2005-11-13 15:18:41.000000000 -0500
+++ balsa-2.3.4/images/Makefile.in	2005-11-13 15:19:21.000000000 -0500
@@ -587,8 +587,8 @@
 # installed in $(SUBDIRS) (Gtk+-2.6)
 install-data-hook:
 	if test -x "$(GTK_UPDATE_ICON_CACHE)" ; then                    \
-		$(GTK_UPDATE_ICON_CACHE) $(DESTDIR)$(BALSA_DATA_PREFIX)/hicolor ; \
-		$(GTK_UPDATE_ICON_CACHE) $(DESTDIR)$(HICOLOR_ICON_DIR) ;          \
+		$(GTK_UPDATE_ICON_CACHE) --ignore-theme-index $(DESTDIR)$(BALSA_DATA_PREFIX)/hicolor ; \
+		$(GTK_UPDATE_ICON_CACHE) --ignore-theme-index $(DESTDIR)$(HICOLOR_ICON_DIR) ;          \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- balsa-2.3.4/images/Makefile.in.orig	2005-11-13 15:18:41.000000000 -0500
+++ balsa-2.3.4/images/Makefile.in	2005-11-13 15:19:21.000000000 -0500
@@ -587,8 +587,8 @@
 # installed in $(SUBDIRS) (Gtk+-2.6)
 install-data-hook:
 	if test -x "$(GTK_UPDATE_ICON_CACHE)" ; then                    \
-		$(GTK_UPDATE_ICON_CACHE) $(DESTDIR)$(BALSA_DATA_PREFIX)/hicolor ; \
-		$(GTK_UPDATE_ICON_CACHE) $(DESTDIR)$(HICOLOR_ICON_DIR) ;          \
+		$(GTK_UPDATE_ICON_CACHE) --ignore-theme-index $(DESTDIR)$(BALSA_DATA_PREFIX)/hicolor ; \
+		$(GTK_UPDATE_ICON_CACHE) --ignore-theme-index $(DESTDIR)$(HICOLOR_ICON_DIR) ;          \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
