Index: tracker-0.6.90/data/trackerd.desktop.in.in
===================================================================
--- tracker-0.6.90.orig/data/trackerd.desktop.in.in	2009-02-16 22:54:34.000000000 +0000
+++ tracker-0.6.90/data/trackerd.desktop.in.in	2009-02-16 22:55:03.000000000 +0000
@@ -12,7 +12,7 @@
 X-KDE-StartupNotify=false
 X-KDE-UniqueApplet=true
 NoDisplay=true
-OnlyShowIn=GNOME;KDE;XFCE;
+OnlyShowIn=GNOME;XFCE;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=tracker
 X-GNOME-Bugzilla-Component=Indexer
