--- browser/app/mozilla.in.old	2008-01-03 18:11:28.000000000 +0000
+++ browser/app/mozilla.in	2008-01-03 18:12:18.000000000 +0000
@@ -95,6 +95,14 @@
 moz_libdir=%MOZAPPDIR%
 MRE_HOME=%MREDIR%
 
+# Include /usr/lib/mozilla/plugins in the plugin path:
+if [ "$MOZ_PLUGIN_PATH" ] ; then
+  MOZ_PLUGIN_PATH=$MOZ_PLUGIN_PATH:${moz_libdir}/plugins:/usr/lib/mozilla/plugins
+else
+  MOZ_PLUGIN_PATH=${moz_libdir}/plugins:/usr/lib/mozilla/plugins
+fi
+export MOZ_PLUGIN_PATH
+
 # Use run-mozilla.sh in the current dir if it exists
 # If not, then start resolving symlinks until we find run-mozilla.sh
 found=0
