--- libvisual/lv_cpu.c_org	2006-03-28 20:25:13.000000000 +0200
+++ libvisual/lv_cpu.c	2006-03-28 20:35:40.000000000 +0200
@@ -163,14 +163,14 @@
 	} else {
 		__lv_powerpc_canjump = 1;
 
-		asm volatile
+		/* asm volatile
 			("mtspr 256, %0\n\t"
 			 "vand %%v0, %%v0, %%v0"
 			 :
-			 : "r" (-1));
+			 : "r" (-1)); */
 
 		signal (SIGILL, SIG_DFL);
-		__lv_cpu_caps.hasAltiVec = 1;
+		__lv_cpu_caps.hasAltiVec = 0;
 	}
 #endif
 }
