--- layout/applications.menu	2008-02-10 22:49:13.000000000 +0000
+++ layout/applications.menu.old	2008-02-19 00:24:43.000000000 +0000
@@ -35,6 +35,8 @@
     </Include>
     <Exclude>
         <Category>KDE</Category>
+        <Category>X-XFCE</Category>
+        <Filename>mousepad.desktop</Filename>
     </Exclude>
   </Menu> <!-- End Accessories -->
 
@@ -52,6 +54,7 @@
     </Include>
     <Exclude>
         <Category>KDE</Category>
+        <Category>X-XFCE</Category>
     </Exclude>
   </Menu> <!-- End Accessibility -->
 
@@ -67,6 +70,7 @@
     </Include>
     <Exclude>
         <Category>KDE</Category>
+        <Category>X-XFCE</Category>
     </Exclude>
   </Menu> <!-- End Development Tools -->
 
@@ -81,6 +85,7 @@
     </Include>
     <Exclude>
         <Category>KDE</Category>
+        <Category>X-XFCE</Category>
     </Exclude>
   </Menu> <!-- End Education -->
 
@@ -95,6 +100,7 @@
     </Include>
     <Exclude>
         <Category>KDE</Category>
+        <Category>X-XFCE</Category>
     </Exclude>
   </Menu> <!-- End Games -->
 
@@ -109,6 +115,7 @@
     </Include>
     <Exclude>
         <Category>KDE</Category>
+        <Category>X-XFCE</Category>
     </Exclude>
   </Menu> <!-- End Graphics -->
 
@@ -123,6 +130,7 @@
     </Include>
     <Exclude>
         <Category>KDE</Category>
+        <Category>X-XFCE</Category>
     </Exclude>
   </Menu>   <!-- End Internet -->
 
@@ -137,6 +145,7 @@
     </Include>
     <Exclude>
         <Category>KDE</Category>
+        <Category>X-XFCE</Category>
     </Exclude>
   </Menu>   <!-- End Multimedia -->
 
@@ -154,6 +163,7 @@
     </Include>
     <Exclude>
         <Category>KDE</Category>
+        <Category>X-XFCE</Category>
     </Exclude>
   </Menu> <!-- End Office -->
 
@@ -169,6 +179,7 @@
     </Include>
     <Exclude>
         <Category>KDE</Category>
+        <Category>X-XFCE</Category>
     </Exclude>
   </Menu>   <!-- End System Tools -->
 
@@ -186,7 +197,23 @@
     </Include>
     <Exclude>
         <Category>KDE</Category>
+        <Category>X-XFCE</Category>
     </Exclude>
   </Menu> <!-- End Other -->
 
+  <Menu>
+     <Name>KDE Applications</Name>
+     <MergeFile>gnome-menus-kde.menu</MergeFile>
+  </Menu>
+
+  <Menu>
+     <Name>OpenOffice.Org</Name>
+     <MergeFile>gnome-menus-ooo.menu</MergeFile>
+  </Menu>
+
+  <Menu>
+     <Name>Xfce Applications</Name>
+     <MergeFile>gnome-menus-xfce.menu</MergeFile>
+  </Menu>
+
 </Menu> <!-- End Applications -->
