diff -Naur opal-3.4.1.old/plugins/video/H.263-1998/h263-1998.cxx opal-3.4.1/plugins/video/H.263-1998/h263-1998.cxx
--- opal-3.4.1.old/plugins/video/H.263-1998/h263-1998.cxx	2008-09-15 12:46:42.000000000 +0100
+++ opal-3.4.1/plugins/video/H.263-1998/h263-1998.cxx	2008-09-23 01:32:13.000000000 +0100
@@ -163,7 +163,8 @@
 
   _context->max_b_frames = 0;
   _context->pix_fmt = PIX_FMT_YUV420P;
-  _context->rtp_mode = 1;
+  // obsoleted
+  //_context->rtp_mode = 1;
 
   // X-Lite does not like Custom Picture frequency clocks...
   _context->time_base.num = 100; 
diff -Naur opal-3.4.1.old/plugins/video/MPEG4-ffmpeg/mpeg4.cxx opal-3.4.1/plugins/video/MPEG4-ffmpeg/mpeg4.cxx
--- opal-3.4.1.old/plugins/video/MPEG4-ffmpeg/mpeg4.cxx	2008-09-15 12:46:45.000000000 +0100
+++ opal-3.4.1/plugins/video/MPEG4-ffmpeg/mpeg4.cxx	2008-09-23 01:32:33.000000000 +0100
@@ -567,7 +567,8 @@
 void MPEG4EncoderContext::SetStaticEncodingParams(){
     _avcontext->pix_fmt = PIX_FMT_YUV420P;
     _avcontext->mb_decision = FF_MB_DECISION_SIMPLE;    // high quality off
-    _avcontext->rtp_mode = 1;                           // use RTP packetization
+    // obsoleted
+    //_avcontext->rtp_mode = 1;                           // use RTP packetization
     _avcontext->rtp_payload_size = 750;                 // ffh263 uses 750
     _avcontext->rtp_callback = &MPEG4EncoderContext::RtpCallback;
 
