diff -Naur ewl-0.0.4.004.orig/src/bin/ewl_test.h ewl-0.0.4.004/src/bin/ewl_test.h
--- ewl-0.0.4.004.orig/src/bin/ewl_test.h	2005-07-10 20:14:53.000000000 -0400
+++ ewl-0.0.4.004/src/bin/ewl_test.h	2005-09-21 17:24:05.000000000 -0400
@@ -1,5 +1,6 @@
 #include "Ewl.h"
 #include <ewl-config.h>
+#include <ewl_macros.h>
 
 #if HAVE___ATTRIBUTE__
 #define __UNUSED__ __attribute__((unused))
diff -Naur ewl-0.0.4.004.orig/src/bin/ewl_theme_test.c ewl-0.0.4.004/src/bin/ewl_theme_test.c
--- ewl-0.0.4.004.orig/src/bin/ewl_theme_test.c	2005-05-29 01:59:23.000000000 -0400
+++ ewl-0.0.4.004/src/bin/ewl_theme_test.c	2005-09-21 17:23:51.000000000 -0400
@@ -1,4 +1,5 @@
 #include "ewl_test.h"
+#include <dirent.h>
 
 static Ewl_Widget *notebook;
 static char *theme_list_dir = PACKAGE_DATA_DIR "/themes/";
diff -Naur ewl-0.0.4.004.orig/src/lib/Ewl.h.in ewl-0.0.4.004/src/lib/Ewl.h.in
--- ewl-0.0.4.004.orig/src/lib/Ewl.h.in	2005-07-10 20:14:53.000000000 -0400
+++ ewl-0.0.4.004/src/lib/Ewl.h.in	2005-09-21 17:23:51.000000000 -0400
@@ -257,6 +257,7 @@
 #include <Evas.h>
 #include <Edje.h>
 #include <Ecore.h>
+#include <Ecore_Data.h>
 #include <Edb.h>
 
 #include <stdio.h>
@@ -272,6 +273,13 @@
 #define PATH_MAX 4096
 #endif
 
+#ifndef MAX
+#define	MAX(a,b) (((a)>(b))?(a):(b))
+#endif
+#ifndef MIN
+#define	MIN(a,b) (((a)<(b))?(a):(b))
+#endif
+
 #include <ewl_enums.h>
 
 #include <ewl_object.h>
diff -Naur ewl-0.0.4.004.orig/src/lib/ewl_config.c ewl-0.0.4.004/src/lib/ewl_config.c
--- ewl-0.0.4.004.orig/src/lib/ewl_config.c	2005-03-28 09:32:26.000000000 -0500
+++ ewl-0.0.4.004/src/lib/ewl_config.c	2005-09-21 17:23:51.000000000 -0400
@@ -210,8 +210,8 @@
 
 		cc = ewl_config_int_get("/ewl/theme/color_classes/count");
 		prop = ecore_config_get("/ewl/theme/color_classes/count");
-		prop->flags &= ~PF_MODIFIED;
-		prop->flags |= PF_SYSTEM;
+		prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+		prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 		for (i = 0; i < cc; i++) {
 			char *name;
@@ -221,8 +221,8 @@
 					"/ewl/theme/color_classes/%d/name", i);
 			name = ewl_config_str_get(key);
 			prop = ecore_config_get(key);
-			prop->flags &= ~PF_MODIFIED;
-			prop->flags |= PF_SYSTEM;
+			prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+			prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 			if (name) {
 				int r, g, b, a;
@@ -233,85 +233,85 @@
 						"/ewl/theme/color_classes/%d/r", i);
 				r = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				snprintf(key, PATH_MAX,
 						"/ewl/theme/color_classes/%d/g", i);
 				g = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				snprintf(key, PATH_MAX,
 						"/ewl/theme/color_classes/%d/b", i);
 				b = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				snprintf(key, PATH_MAX,
 						"/ewl/theme/color_classes/%d/a", i);
 				a = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				snprintf(key, PATH_MAX,
 						"/ewl/theme/color_classes/%d/r2", i);
 				r2 = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				snprintf(key, PATH_MAX,
 						"/ewl/theme/color_classes/%d/g2", i);
 				g2 = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				snprintf(key, PATH_MAX,
 						"/ewl/theme/color_classes/%d/b2", i);
 				b2 = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				snprintf(key, PATH_MAX,
 						"/ewl/theme/color_classes/%d/a2", i);
 				a2 = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				snprintf(key, PATH_MAX,
 						"/ewl/theme/color_classes/%d/r3", i);
 				r3 = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				snprintf(key, PATH_MAX,
 						"/ewl/theme/color_classes/%d/g3", i);
 				g3 = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				snprintf(key, PATH_MAX,
 						"/ewl/theme/color_classes/%d/b3", i);
 				b3 = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				snprintf(key, PATH_MAX,
 						"/ewl/theme/color_classes/%d/a3", i);
 				a3 = ewl_config_int_get(key);
 				prop = ecore_config_get(key);
-				prop->flags &= ~PF_MODIFIED;
-				prop->flags |= PF_SYSTEM;
+				prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+				prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 
 				edje_color_class_set(name, r, g, b, a,
 						r2, g2, b2, a2,
@@ -389,8 +389,8 @@
 
 		for(i = 0; keys[i] != NULL; i++) {
 			prop = ecore_config_get(keys[i]);
-			prop->flags &= ~PF_MODIFIED;
-			prop->flags |= PF_SYSTEM;
+			prop->flags &= ~ECORE_CONFIG_FLAG_MODIFIED;
+			prop->flags |= ECORE_CONFIG_FLAG_SYSTEM;
 		}
 
 		ecore_config_listen("ewl_debug_enable", "/ewl/debug/enable",
