--- configure.in.orig	2005-02-09 03:12:02.000000000 -0500
+++ configure.in	2005-02-09 03:12:43.000000000 -0500
@@ -67,7 +67,7 @@
         [HAVE_FPOS_T_POS])
 
 
-external_mp4v2=yes
+AC_CHECK_PROG(external_mp4v2, mpeg4ip-config, no, no)
 AC_CHECK_LIB(mp4v2, MP4Create, , external_mp4v2=no, -lstdc++)
 AC_CHECK_LIB(mp4v2, MP4MetadataDelete, , external_mp4v2=no, -lstdc++)
 AC_CHECK_HEADER(mp4.h, , external_mp4v2=no)
