diff -Naur evolution-caldav-2.3.99.3.orig/configure evolution-caldav-2.3.99.3/configure
--- evolution-caldav-2.3.99.3.orig/configure	2005-12-01 18:03:52.000000000 -0500
+++ evolution-caldav-2.3.99.3/configure	2006-05-04 12:02:50.000000000 -0400
@@ -1988,7 +1988,7 @@
           ac_config_headers="$ac_config_headers config.h"
 
 
-EVOLUTION_API_VERSION=2.4
+EVOLUTION_API_VERSION=2.6
 EDS_API_VERSION=1.2
 
 echo "$as_me:$LINENO: checking Evolution version" >&5
