diff -Naur libgnomeprintui-2.12.1.orig/configure libgnomeprintui-2.12.1/configure
--- libgnomeprintui-2.12.1.orig/configure	2005-09-22 15:37:29.000000000 -0400
+++ libgnomeprintui-2.12.1/configure	2005-10-10 00:04:30.000000000 -0400
@@ -22702,13 +22702,11 @@
 	gtk+-2.0	   >= 2.6.0
 	libgnomeprint-2.2  >= 2.12.1
 	libgnomecanvas-2.0 >= 1.117.0
-	gnome-icon-theme   >= 1.1.92
 \"") >&5
   ($PKG_CONFIG --exists --print-errors "
 	gtk+-2.0	   >= 2.6.0
 	libgnomeprint-2.2  >= 2.12.1
 	libgnomecanvas-2.0 >= 1.117.0
-	gnome-icon-theme   >= 1.1.92
 ") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
@@ -22717,7 +22715,6 @@
 	gtk+-2.0	   >= 2.6.0
 	libgnomeprint-2.2  >= 2.12.1
 	libgnomecanvas-2.0 >= 1.117.0
-	gnome-icon-theme   >= 1.1.92
 " 2>/dev/null`
 else
   pkg_failed=yes
@@ -22731,13 +22728,11 @@
 	gtk+-2.0	   >= 2.6.0
 	libgnomeprint-2.2  >= 2.12.1
 	libgnomecanvas-2.0 >= 1.117.0
-	gnome-icon-theme   >= 1.1.92
 \"") >&5
   ($PKG_CONFIG --exists --print-errors "
 	gtk+-2.0	   >= 2.6.0
 	libgnomeprint-2.2  >= 2.12.1
 	libgnomecanvas-2.0 >= 1.117.0
-	gnome-icon-theme   >= 1.1.92
 ") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
@@ -22746,7 +22741,6 @@
 	gtk+-2.0	   >= 2.6.0
 	libgnomeprint-2.2  >= 2.12.1
 	libgnomecanvas-2.0 >= 1.117.0
-	gnome-icon-theme   >= 1.1.92
 " 2>/dev/null`
 else
   pkg_failed=yes
@@ -22760,7 +22754,6 @@
 	gtk+-2.0	   >= 2.6.0
 	libgnomeprint-2.2  >= 2.12.1
 	libgnomecanvas-2.0 >= 1.117.0
-	gnome-icon-theme   >= 1.1.92
 "`
 	# Put the nasty error message in config.log where it belongs
 	echo "$LIBGNOMEPRINTUI_PKG_ERRORS" 1>&5
@@ -22769,7 +22762,6 @@
 	gtk+-2.0	   >= 2.6.0
 	libgnomeprint-2.2  >= 2.12.1
 	libgnomecanvas-2.0 >= 1.117.0
-	gnome-icon-theme   >= 1.1.92
 ) were not met.
 Consider adjusting the PKG_CONFIG_PATH environment variable if you
 installed software in a non-standard prefix.
@@ -22781,7 +22773,6 @@
 	gtk+-2.0	   >= 2.6.0
 	libgnomeprint-2.2  >= 2.12.1
 	libgnomecanvas-2.0 >= 1.117.0
-	gnome-icon-theme   >= 1.1.92
 ) were not met.
 Consider adjusting the PKG_CONFIG_PATH environment variable if you
 installed software in a non-standard prefix.
