--- libdvdcss-1.2.9/Makefile.in.orig	2005-07-12 02:12:30.000000000 +1000
+++ libdvdcss-1.2.9/Makefile.in	2006-06-30 08:25:31.905022750 +1000
@@ -191,7 +191,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-SUBDIRS = src test doc
+SUBDIRS = src test
 DIST_SUBDIRS = $(SUBDIRS) debian
 EXTRA_DIST = libdvdcss.spec bootstrap
 AUTOMAKE_OPTIONS = foreign dist-bzip2 subdir-objects
