diff -Naur seahorse-2.26.1.old/daemon/Makefile.am seahorse-2.26.1/daemon/Makefile.am
--- seahorse-2.26.1.old/daemon/Makefile.am	2009-02-03 18:09:54.000000000 +0000
+++ seahorse-2.26.1/daemon/Makefile.am	2009-05-04 21:34:27.000000000 +0100
@@ -14,7 +14,7 @@
 	-DLOCALEDIR=\""$(localedir)"\" \
 	-DPIXMAPSDIR=\""$(pixmapsdir)"\" \
 	-DLIBCRYPTUI_API_SUBJECT_TO_CHANGE \
-	-DAUTOSTARTDIR=\""$(sysconfdir)/xdg/autostart"\"
+	-DAUTOSTARTDIR=\""$(sysconfdir)/gnome/xdg/autostart"\"
 
 bin_PROGRAMS = seahorse-daemon
 
@@ -66,7 +66,7 @@
 glade_DATA = 
 
 #autostart
-desktopdir = $(sysconfdir)/xdg/autostart
+desktopdir = $(sysconfdir)/gnome/xdg/autostart
 desktop_in_files = seahorse-daemon.desktop.in 
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 @INTLTOOL_DESKTOP_RULE@
diff -Naur seahorse-2.26.1.old/daemon/Makefile.in seahorse-2.26.1/daemon/Makefile.in
--- seahorse-2.26.1.old/daemon/Makefile.in	2009-04-12 15:33:22.000000000 +0100
+++ seahorse-2.26.1/daemon/Makefile.in	2009-05-04 21:34:40.000000000 +0100
@@ -306,7 +306,7 @@
 	-DLOCALEDIR=\""$(localedir)"\" \
 	-DPIXMAPSDIR=\""$(pixmapsdir)"\" \
 	-DLIBCRYPTUI_API_SUBJECT_TO_CHANGE \
-	-DAUTOSTARTDIR=\""$(sysconfdir)/xdg/autostart"\"
+	-DAUTOSTARTDIR=\""$(sysconfdir)/gnome/xdg/autostart"\"
 
 BUILT_SOURCES = \
 	seahorse-service-bindings.h \
@@ -342,7 +342,7 @@
 glade_DATA = 
 
 #autostart
-desktopdir = $(sysconfdir)/xdg/autostart
+desktopdir = $(sysconfdir)/gnome/xdg/autostart
 desktop_in_files = seahorse-daemon.desktop.in 
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
