--- dbus_service.c	2007-08-13 18:38:27.000000000 +0100
+++ /root/dbus_service.c.patched	2007-10-14 17:12:27.000000000 +0100
@@ -1024,7 +1024,12 @@
 
 static void set_watch_fds(DBusWatch * watch, DBusConnectionState * cs) {
     uint8_t flags = dbus_watch_get_flags(watch);
+    #if (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MINOR == 1 && DBUS_VERSION_MICRO >= 1) || (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MAJOR > 1) || (DBUS_VERSION_MAJOR > 1)
+
     int fd = dbus_watch_get_unix_fd(watch);
+#else
+    int fd = dbus_watch_get_fd(watch);
+#endif
 
     if (cs->n <= fd)
         cs->n = fd + 1;
@@ -1057,7 +1062,13 @@
     dbus_watch_set_data(watch, cs, no_free);
 
     if (cs->dh != 0L)
-        (*(cs->dh)) ("add_watch: %d", dbus_watch_get_unix_fd (watch));
+        (*(cs->dh)) ("add_watch: %d", 
+
+#if (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MINOR == 1 && DBUS_VERSION_MICRO >= 1) || (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MAJOR > 1) || (DBUS_VERSION_MAJOR > 1)
+    dbus_watch_get_unix_fd (watch));
+#else
+    dbus_watch_get_fd (watch));
+#endif;
 
     if (tsearch((const void *) watch, &(cs->watches), ptr_key_comparator) == 0L) {
         if (cs->eh != 0L)
@@ -1074,14 +1085,25 @@
 static void remove_watch(DBusWatch * watch, void *csp)
 {
     DBusConnectionState *cs = csp;
+
+#if (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MINOR == 1 && DBUS_VERSION_MICRO >= 1) || (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MAJOR > 1) || (DBUS_VERSION_MAJOR > 1)
     int fd = dbus_watch_get_unix_fd(watch);
+#else
+    int fd = dbus_watch_get_fd(watch);
+#endif
 
     if (tdelete((const void *) watch, &(cs->watches), ptr_key_comparator) == 0L)
         if (cs->eh != 0L)
             (*(cs->eh)) ("remove_watch: can't happen?!?: watch not found");
 
     if (cs->dh != 0L)
-        (*(cs->dh)) ("remove_watch: %d", dbus_watch_get_unix_fd(watch));
+        (*(cs->dh)) ("remove_watch: %d", 
+
+#if (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MINOR == 1 && DBUS_VERSION_MICRO >= 1) || (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MAJOR > 1) || (DBUS_VERSION_MAJOR > 1)
+        dbus_watch_get_unix_fd(watch));
+#else
+        dbus_watch_get_fd(watch));
+#endif
 
     FD_CLR(fd, &(cs->r_fds));
     FD_CLR(fd, &(cs->w_fds));
@@ -1092,7 +1114,13 @@
     DBusConnectionState *cs = csp;
 
     if (cs->dh != 0L)
-        (*(cs->dh)) ("toggle_watch: %d", dbus_watch_get_unix_fd(watch));
+        (*(cs->dh)) ("toggle_watch: %d", 
+        
+#if (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MINOR == 1 && DBUS_VERSION_MICRO >= 1) || (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MAJOR > 1) || (DBUS_VERSION_MAJOR > 1)
+        dbus_watch_get_unix_fd(watch));
+#else
+        dbus_watch_get_fd(watch));
+#endif
 
     set_watch_fds(watch, cs);
 }
@@ -1114,11 +1142,21 @@
         if (!dbus_watch_get_enabled(w))
             return;
 
+#if (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MINOR == 1 && DBUS_VERSION_MICRO >= 1) || (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MAJOR > 1) || (DBUS_VERSION_MAJOR > 1)
         fd = dbus_watch_get_unix_fd(w);
+#else
+        fd = dbus_watch_get_fd(w);
+#endif
         flags = dbus_watch_get_flags(w);
 
         if (cs->dh != 0L)
-            (*(cs->dh)) ("handle_watch: %d", dbus_watch_get_unix_fd(w));
+            (*(cs->dh)) ("handle_watch: %d", 
+#if (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MINOR == 1 && DBUS_VERSION_MICRO >= 1) || (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MAJOR > 1) || (DBUS_VERSION_MAJOR > 1)
+        dbus_watch_get_unix_fd(w));
+#else
+        dbus_watch_get_fd(w));
+#endif
+
 
         if ((flags & DBUS_WATCH_READABLE) && (FD_ISSET(fd, &(cs->s_r_fds))))
             dbus_watch_handle(w, DBUS_WATCH_READABLE);
