--- libgnome-2.10.0/schemas/desktop_gnome_background.schemas.in.orig	2005-03-14 16:27:01.000000000 -0500
+++ libgnome-2.10.0/schemas/desktop_gnome_background.schemas.in	2005-03-14 16:29:37.000000000 -0500
@@ -16,7 +16,7 @@
         <applyto>/desktop/gnome/background/picture_options</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>wallpaper</default>
+        <default>stretched</default>
         <locale name="C">
         <short>Picture Options</short>
         <long>Determines how the image set by wallpaper_filename is
@@ -29,7 +29,7 @@
         <applyto>/desktop/gnome/background/picture_filename</applyto>
         <owner>gnome</owner>
         <type>string</type>
-	<default></default>
+	<default>/usr/share/pixmaps/backgrounds/frg/frg.png</default>
         <locale name="C">
         <short>Picture Filename</short>
         <long>File to use for the background image</long>
