--- libgnome-2.10.0/schemas/desktop_gnome_peripherals_mouse.schemas.in.orig	2005-04-18 15:07:37.000000000 -0400
+++ libgnome-2.10.0/schemas/desktop_gnome_peripherals_mouse.schemas.in	2005-04-18 15:08:26.000000000 -0400
@@ -100,7 +100,7 @@
     <schema>
       <key>/schemas/desktop/gnome/peripherals/mouse/cursor_theme</key>
       <applyto>/desktop/gnome/peripherals/mouse/cursor_theme</applyto>
-      <owner>gnome</owner>
+      <owner>jimmac</owner>
       <type>string</type>
       <locale name="C">
 	<short>Cursor theme</short>
