--- configure.in.old
+++ configure.in
@@ -1114,24 +1114,40 @@
     AC_SUBST(GLADE_LIBS)
 
-    PKG_CHECK_MODULES(GOFFICE, libgoffice-0.4 >= 0.4.0, [], [
-      PKG_CHECK_MODULES(GOFFICE, libgoffice-0.3 >= 0.3.0, [], [
-        PKG_CHECK_MODULES(GOFFICE, libgoffice-1 >= 0.0.4, [], [
-          AC_MSG_ERROR([Cannot find libgoffice.])
+    # checks for goffice
+    goffice=0
+    goffice_with_cairo=0
+    PKG_CHECK_MODULES(GOFFICE, libgoffice-0.5 >= 0.5.1, [
+      AC_DEFINE(HAVE_GOFFICE_0_5,1,[System has goffice 0.5.1 or better])
+      AC_DEFINE(GOFFICE_WITH_CAIRO,1,[GOffice has been built with cairo support])
+      goffice=1
+      goffice_with_cairo=1
+    ], [goffice=0])
+
+    if test x$goffice = x0
+    then
+      PKG_CHECK_MODULES(GOFFICE, libgoffice-0.4 >= 0.4.0, [goffice=1], [
+        PKG_CHECK_MODULES(GOFFICE, libgoffice-0.3 >= 0.3.0, [goffice=1], [
+          PKG_CHECK_MODULES(GOFFICE, libgoffice-1 >= 0.0.4, [goffice=1], [
+            AC_MSG_ERROR([Cannot find libgoffice.])
+          ])
         ])
       ])
-    ])
+    fi
     AS_SCRUB_INCLUDE(GOFFICE_CFLAGS)
     AC_SUBST(GOFFICE_CFLAGS)
     AC_SUBST(GOFFICE_LIBS)
 
-    saved_CPPFLAGS="${CPPFLAGS}"
-    CPPFLAGS="${GOFFICE_CFLAGS} ${CPPFLAGS}"
-    AC_CHECK_HEADER(goffice/graph/gog-renderer-cairo.h, [
-      AC_DEFINE(GOFFICE_WITH_CAIRO,1,[GOffice has been built with cairo support])
-      goffice_with_cairo=yes
-    ], [
-      goffice_with_cairo=no
-    ])
-    CPPFLAGS="${saved_CPPFLAGS}"
+    if test x$goffice_with_cairo = x0
+    then
+      saved_CPPFLAGS="${CPPFLAGS}"
+      CPPFLAGS="${GOFFICE_CFLAGS} ${CPPFLAGS}"
+      AC_CHECK_HEADER(goffice/graph/gog-renderer-cairo.h, [
+        AC_DEFINE(GOFFICE_WITH_CAIRO,1,[GOffice has been built with cairo support])
+        goffice_with_cairo=1
+      ], [
+        goffice_with_cairo=0
+      ])
+      CPPFLAGS="${saved_CPPFLAGS}"
+    fi
 
     # check for gtkhtml >= 3.14 with gtkprint support
@@ -1140,5 +1156,5 @@
       if test "x$HAVE_GTK_2_10" != "xyes"; then
         AC_MSG_NOTICE([Found libgtkhtml-3.14, but not gtk+-2.0 >= 2.10])
-      elif test "x$goffice_with_cairo" != "xyes"; then
+      elif test "x$goffice_with_cairo" != "x1"; then
         AC_MSG_NOTICE([Found libgtkhtml-3.14, but goffice lacks a cairo renderer])
       else
