--- gnc-html-graph-gog.c.old
+++ gnc-html-graph-gog.c
@@ -37,5 +37,7 @@
 #include <goffice/graph/gog-graph.h>
 #include <goffice/graph/gog-object.h>
-#ifdef GOFFICE_WITH_CAIRO
+#if defined(HAVE_GOFFICE_0_5)
+#    include <goffice/graph/gog-renderer.h>
+#elif defined(GOFFICE_WITH_CAIRO)
 #    include <goffice/graph/gog-renderer-cairo.h>
 #else
@@ -171,5 +173,7 @@
 {
   GtkWidget *widget;
-#ifdef GOFFICE_WITH_CAIRO
+#if defined(HAVE_GOFFICE_0_5)
+  GogRenderer *renderer;
+#elif defined(GOFFICE_WITH_CAIRO)
   GogRendererCairo *cairo_renderer;
 #else
@@ -185,5 +189,11 @@
   gog_object_update (GOG_OBJECT (graph));
 
-#ifdef GOFFICE_WITH_CAIRO
+#if defined(HAVE_GOFFICE_0_5)
+  renderer = GOG_RENDERER (g_object_new (GOG_RENDERER_TYPE,
+                    "model", graph,
+                    NULL));
+  update_status = gog_renderer_update (renderer, eb->width, eb->height);
+  buf = gog_renderer_get_pixbuf (renderer);
+#elif defined(GOFFICE_WITH_CAIRO)
   cairo_renderer = GOG_RENDERER_CAIRO (g_object_new (GOG_RENDERER_CAIRO_TYPE,
                             "model", graph,
@@ -199,5 +209,5 @@
                          eb->width, eb->height, 1.0);
   buf = gog_renderer_pixbuf_get (pixbuf_renderer);
-#endif /* GOFFICE_WITH_CAIRO */
+#endif
 
   widget = gtk_image_new_from_pixbuf (buf);
@@ -629,6 +639,10 @@
 {
   GogGraph *graph = GOG_GRAPH(g_object_get_data(G_OBJECT(eb), "graph"));
+#    ifdef HAVE_GOFFICE_0_5
+  GogRenderer *rend = g_object_new(GOG_RENDERER_TYPE, "model", graph, NULL);
+#    else
   GogRendererCairo *rend = g_object_new(GOG_RENDERER_CAIRO_TYPE, "model", graph,
                                         "cairo", cr, "is-vector", TRUE, NULL);
+#    endif
 
   /* assuming pixel size is 0.5, cf. gtkhtml/src/htmlprinter.c */
@@ -636,5 +650,10 @@
 
   cairo_translate(cr, 0, -eb->height);
+
+#    ifdef HAVE_GOFFICE_0_5
+  gog_renderer_render_to_cairo(rend, cr, eb->width, eb->height);
+#    else
   gog_renderer_cairo_update(rend, eb->width, eb->height, 1.0);
+#    endif
   g_object_unref(rend);
 }
