--- gnome-session-2.10.0/gnome-session/gnome-session.schemas.in.orig	2005-03-20 12:21:23.000000000 -0500
+++ gnome-session-2.10.0/gnome-session/gnome-session.schemas.in	2005-03-20 12:22:18.000000000 -0500
@@ -50,7 +50,7 @@
          <applyto>/apps/gnome-session/options/splash_image</applyto>
          <owner>gnome</owner>
          <type>string</type>
-         <default>splash/gnome-splash.png</default>
+         <default>splash/frg/frg_darkerII_splash_without_bar.png</default>
          <locale name="C">
             <short>Preferred Image to use for login splash screen</short>
             <long>This is a relative path value based off the $datadir/pixmaps/ directory.  Sub-directories and image names are valid values. Changing this value will effect the next session login.</long>
