--- gnome-session/default.in.orig	Wed Aug 10 23:50:50 2005
+++ gnome-session/default.in	Wed Aug 10 23:52:14 2005
@@ -8,7 +8,7 @@
 # The clients must be numbered from 0 to the value of num_clients - 1.
 
 [Default]
-num_clients=4
+num_clients=5
 0,id=default0
 0,Priority=0
 0,RestartCommand=gnome-smproxy --sm-client-id default0
@@ -21,3 +21,6 @@
 3,id=default3
 3,Priority=40
 3,RestartCommand=nautilus --no-default-window --sm-client-id default3
+4,id=default4
+4,Priority=50
+4,RestartCommand=gnome-volume-manager --sm-client-id default4
