--- gnome-blog-0.8/gnomeblog.schemas.in.orig	2005-03-17 15:04:28.000000000 -0500
+++ gnome-blog-0.8/gnomeblog.schemas.in	2005-03-17 15:04:59.000000000 -0500
@@ -8,7 +8,7 @@
       <applyto>/apps/gnome-blog/initialized</applyto>
       <owner>gnome-blog</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
          <short>The application's preferences have been initialized</short>
       </locale>
