diff -up gvfs-1.2.1/programs/Makefile.am.bash-completion gvfs-1.2.1/programs/Makefile.am
--- gvfs-1.2.1/programs/Makefile.am.bash-completion	2009-04-11 00:36:25.730330858 -0400
+++ gvfs-1.2.1/programs/Makefile.am	2009-04-11 00:36:37.746330659 -0400
@@ -34,7 +34,7 @@ bin_SCRIPTS =					\
 
 if ENABLE_BASHCOMP
 profiledir = $(BASHCOMP_DIR)
-profile_SCRIPTS = gvfs-bash-completion.sh
+profile_DATA = gvfs-bash-completion.sh
 endif
 
 gvfs_cat_SOURCES = gvfs-cat.c
