--- header.html.old	2007-12-10 17:28:54.000000000 +0000
+++ header.html	2007-12-10 17:32:26.000000000 +0000
@@ -1,7 +1,7 @@
 <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
 <html>
   <head>
-    <meta http-equiv="Content-Type" content="text/html;charset=iso-8859-1">
+    <meta http-equiv="Content-Type" content="text/html;charset=utf-8">
     <title>libdvdcss documentation</title>
     <link href="/main.css" rel="stylesheet" type="text/css">
     <link href="/doxygen.css" rel="stylesheet" type="text/css">
--- doxygen.cfg.old	2007-12-10 17:28:45.000000000 +0000
+++ doxygen.cfg	2007-12-10 17:32:17.000000000 +0000
@@ -364,6 +364,10 @@
 INPUT                  = ../src/dvdcss/dvdcss.h \
                          ../src/libdvdcss.c
 
+# The INPUT_ENCODING tag sets the character encoding of the input files.
+
+INPUT_ENCODING         = ISO-8859-1
+
 # If the value of the INPUT tag contains directories, you can use the 
 # FILE_PATTERNS tag to specify one or more wildcard pattern (like *.cpp 
 # and *.h) to filter out the source-files in the directories. If left 
